diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/useraccounts mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/useraccounts
--- e-smith-base-4.18.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/useraccounts	2007-01-19 14:33:22.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/useraccounts	2007-10-14 09:01:44.000000000 -0600
@@ -398,15 +398,12 @@
         <trans>
             <![CDATA[
             The system password has been changed.
-            Since this manager application is password protected, you will
-            <b>immediately</b>
-            be prompted for the new system password if you try to continue.
             ]]>
         </trans>
     </entry>
     <entry>
         <base>SYSTEM_PASSWORD_CHANGE_SUCCEEDED</base>
-        <trans>Thse system password has been changed</trans>
+        <trans>The system password has been changed</trans>
     </entry>
     <entry>
         <base>CURRENT_SYSTEM_PASSWORD</base>
