--- e-smith-base-5.0.0/root/usr/share/hal/fdi/95userpolicy/usb-rev70-drive.fdi.usbRev70drive	2009-02-14 12:02:44.000000000 +0100
+++ e-smith-base-5.0.0/root/usr/share/hal/fdi/95userpolicy/usb-rev70-drive.fdi	2009-02-14 11:59:51.000000000 +0100
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="ISO-8859-1"?> <!-- -*- SGML -*- -->
+<deviceinfo version="0.2">
+  <device>
+      <match key="storage.bus" string="usb">
+        <match key="storage.vendor" string="Iomega">
+          <match key="storage.model" string="RRD2">
+            <merge key="storage.policy.desired_mount_point" type="string">usbdisk</merge>
+          </match>
+        </match>
+      </match>
+  </device>
+</deviceinfo>
