diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/sysconfig/network-scripts/ifcfg-ethX/10HWADDR.fix-template-expansion-error e-smith-base-5.2.0/root/etc/e-smith/templates/etc/sysconfig/network-scripts/ifcfg-ethX/10HWADDR
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/sysconfig/network-scripts/ifcfg-ethX/10HWADDR.fix-template-expansion-error	2010-02-14 20:26:17.000000000 +0100
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/sysconfig/network-scripts/ifcfg-ethX/10HWADDR	2010-02-14 20:30:01.000000000 +0100
@@ -1,11 +1,9 @@
 {
     $OUT = "";
     return if ($InternalInterface{Name} eq "bond0");
-    if ($THIS_DEVICE =~ /^eth[01]$/)
-    {
-        $OUT .= "HWADDR=" . (
-            ($is_internal) ?
-            $InternalInterface{HWAddress} : $ExternalInterface{HWAddress}
-        );
-    }
+    return unless $THIS_DEVICE =~ /^eth[01]$/;
+    my $hwaddr = $is_internal ? 
+        $InternalInterface{HWAddress}:
+        $ExternalInterface{HWAddress};
+    $OUT .= "HWADDR=$hwaddr" if $hwaddr;
 }
