diff -up e-smith-base-5.2.0/createlinks.cpu-conf e-smith-base-5.2.0/createlinks
--- e-smith-base-5.2.0/createlinks.cpu-conf	2010-11-01 07:55:51.000000000 -0600
+++ e-smith-base-5.2.0/createlinks	2010-11-01 07:58:01.000000000 -0600
@@ -214,6 +214,14 @@ foreach my $file (qw(
     templates2events($file, qw(console-save bootstrap-console-save));
 }
 
+foreach my $file (qw(
+       /etc/cpu.conf
+       /etc/cpu-system.conf
+    ))
+{
+    templates2events($file, qw(console-save bootstrap-console-save ldap-update));
+}
+
 # conf-userlists
 
 foreach (qw(users.allow accounts.allow accounts.deny))
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/00header.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/00header
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/00header.cpu-conf	2010-11-01 07:58:29.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/00header	2010-11-01 07:58:43.000000000 -0600
@@ -0,0 +1,3 @@
+[GLOBAL]
+DEFAULT_METHOD  = ldap
+
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/05ldap.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/05ldap
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/05ldap.cpu-conf	2010-11-01 07:58:51.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/05ldap	2010-11-01 07:59:03.000000000 -0600
@@ -0,0 +1,4 @@
+[LDAP]
+LDAP_HOST       = localhost
+LDAP_PORT       = 389
+
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/10bindpw.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/10bindpw
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/10bindpw.cpu-conf	2010-11-01 07:59:18.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/10bindpw	2010-11-01 07:59:34.000000000 -0600
@@ -0,0 +1,2 @@
+BIND_DN         = "cn=root,{ esmith::util::ldapBase ($DomainName); }"
+BIND_PASS       = { esmith::util::LdapPassword (); }
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/15defaultObjectClass.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/15defaultObjectClass
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/15defaultObjectClass.cpu-conf	2010-11-01 08:00:02.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/15defaultObjectClass	2010-11-01 08:00:19.000000000 -0600
@@ -0,0 +1,10 @@
+{
+
+our @userObjectClass = qw(inetOrgPerson posixAccount shadowAccount);
+our @systemUserObjectClass =qw(account posixAccount shadowAccount);
+our @groupObjectClass = qw(posixGroup mailboxRelatedObject);
+our @systemGroupObjectClass = qw(posixGroup);
+
+$OUT .= "";
+
+}
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/20objectClass.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/20objectClass
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/20objectClass.cpu-conf	2010-11-01 08:00:27.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/20objectClass	2010-11-01 08:01:06.000000000 -0600
@@ -0,0 +1,15 @@
+{
+if ($ACCOUNT eq 'system'){
+    $OUT .= "USER_OBJECT_CLASS = " . join(",",@systemUserObjectClass) . "\n";
+    $OUT .= "GROUP_OBJECT_CLASS = " . join(",",@systemGroupObjectClass) . "\n";
+}
+else{
+    $OUT .= "USER_OBJECT_CLASS  = " . join(",",@userObjectClass) . "\n";
+    $OUT .= "GROUP_OBJECT_CLASS = " . join(",",@groupObjectClass) . "\n";
+
+}
+
+$OUT .= "USER_BASE = ou=Users," . esmith::util::ldapBase ($DomainName) . "\n";
+$OUT .= "GROUP_BASE = ou=Groups," . esmith::util::ldapBase ($DomainName) ."\n";
+
+}
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/25filter.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/25filter
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/25filter.cpu-conf	2010-11-01 08:01:25.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/25filter	2010-11-01 08:01:37.000000000 -0600
@@ -0,0 +1,2 @@
+USER_FILTER     = (objectClass=posixAccount)
+GROUP_FILTER    = (objectClass=posixGroup)
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/30cnString.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/30cnString
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/30cnString.cpu-conf	2010-11-01 08:01:42.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/30cnString	2010-11-01 08:01:52.000000000 -0600
@@ -0,0 +1,2 @@
+USER_CN_STRING  = uid
+GROUP_CN_STRING = cn
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/35skel.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/35skel
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/35skel.cpu-conf	2010-11-01 08:02:06.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/35skel	2010-11-01 08:02:17.000000000 -0600
@@ -0,0 +1 @@
+SKEL_DIR        = /etc/e-smith/skel/user
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/40shell.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/40shell
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/40shell.cpu-conf	2010-11-01 08:02:23.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/40shell	2010-11-01 08:02:32.000000000 -0600
@@ -0,0 +1 @@
+DEFAULT_SHELL   = /usr/bin/rssh
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/45homeDir.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/45homeDir
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/45homeDir.cpu-conf	2010-11-01 08:02:43.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/45homeDir	2010-11-01 08:02:51.000000000 -0600
@@ -0,0 +1 @@
+HOME_DIRECTORY  = /home/e-smith/files/users
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/50uidGid.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/50uidGid
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/50uidGid.cpu-conf	2010-11-01 08:02:59.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/50uidGid	2010-11-01 08:03:13.000000000 -0600
@@ -0,0 +1,8 @@
+MAX_UIDNUMBER = 10000
+MIN_UIDNUMBER = 100
+MAX_GIDNUMBER = 10000
+MIN_GIDNUMBER = 101
+ID_MAX_PASSES = 1000
+USERGROUPS = yes
+USERS_GID = 100
+RANDOM = "false"
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/55passwdShadow.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/55passwdShadow
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/55passwdShadow.cpu-conf	2010-11-01 08:03:43.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/55passwdShadow	2010-11-01 08:03:56.000000000 -0600
@@ -0,0 +1,2 @@
+PASSWORD_FILE = "/etc/passfile"
+SHADOW_FILE = "/etc/shadowfile"
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/60hash.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/60hash
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/60hash.cpu-conf	2010-11-01 08:04:04.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/60hash	2010-11-01 08:04:15.000000000 -0600
@@ -0,0 +1 @@
+HASH = "crypt"
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/65shadow.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/65shadow
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/65shadow.cpu-conf	2010-11-01 08:04:23.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates/etc/cpu.conf/65shadow	2010-11-01 08:04:35.000000000 -0600
@@ -0,0 +1,7 @@
+SHADOWLASTCHANGE        = 11192
+SHADOWMAX               = 99999
+SHADOWWARING            = 7
+SHADOWEXPIRE            = -1
+SHADOWFLAG              = 134538308
+SHADOWMIN               = -1
+SHADOWINACTIVE          = -1
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/cpu.conf.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/cpu.conf
--- e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/cpu.conf.cpu-conf	2010-11-01 08:10:45.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/cpu.conf	2010-11-01 08:10:55.000000000 -0600
@@ -0,0 +1,2 @@
+PERMS=0600
+MORE_DATA={ ACCOUNT => "user" }
diff -up e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/cpu-system.conf.cpu-conf e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/cpu-system.conf
--- e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/cpu-system.conf.cpu-conf	2010-11-01 08:11:06.000000000 -0600
+++ e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/cpu-system.conf	2010-11-01 08:11:20.000000000 -0600
@@ -0,0 +1,4 @@
+PERMS=0600
+TEMPLATE_PATH="/etc/cpu.conf"
+OUTPUT_FILENAME="/etc/cpu-system.conf"
+MORE_DATA={ ACCOUNT => "system" }
