diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/pam.d/system-auth/40password mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/pam.d/system-auth/40password
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/pam.d/system-auth/40password	2007-05-24 13:06:20.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/pam.d/system-auth/40password	2007-05-24 13:06:04.000000000 -0600
@@ -1,4 +1,3 @@
-password    requisite     pam_cracklib.so retry=3 type=
-password    sufficient    pam_unix.so nullok use_authtok md5 shadow
+password    sufficient    pam_unix.so nullok md5 shadow
 password    required      pam_deny.so
 
