diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/web/functions/localnetworks mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/web/functions/localnetworks
--- e-smith-base-4.18.0/root/etc/e-smith/web/functions/localnetworks	2008-03-12 13:37:46.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/web/functions/localnetworks	2008-03-12 13:35:57.000000000 -0600
@@ -61,7 +61,6 @@
 is  (mode($panel), '4750',              "Check permissions on script");
 ok  ($ua->get_panel($panel),            "ABOUT TO RUN L10N TESTS");
 is  ($ua->{status}, 200,                "200 OK");
-#like($ua->{content}, qr/FORM_TITLE/,   "Saw untranslated form title");
 ok  ($ua->set_language("en-us"),        "Set language to U.S. English");
 ok  ($ua->get_panel($panel),            "Get panel");
 is  ($ua->{status}, 200,                "200 OK");
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/web/functions/review mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/web/functions/review
--- e-smith-base-4.18.0/root/etc/e-smith/web/functions/review	2008-03-12 13:37:46.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/web/functions/review	2008-03-12 13:36:05.000000000 -0600
@@ -56,7 +56,6 @@
 is  (getgrgid($stat[5]), 'admin', "File is owned by group admin");
 ok  ($ua->get_panel($panel),            "ABOUT TO RUN L10N TESTS");
 is  ($ua->{status}, 200,                "200 OK");
-like($ua->{content}, qr/FORM_TITLE/,    "Saw untranslated form title");
 ok  ($ua->set_language("en"),           "Set language to English");
 ok  ($ua->get_panel($panel),            "Get panel");
 is  ($ua->{status}, 200,                "200 OK");
