diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/createlinks mezzanine_patched_e-smith-base-5.2.0/createlinks
--- e-smith-base-5.2.0/createlinks	2010-09-23 16:27:49.000000000 +0200
+++ mezzanine_patched_e-smith-base-5.2.0/createlinks	2010-09-23 16:27:31.000000000 +0200
@@ -206,6 +206,8 @@
 	/etc/pam.d/system-auth
 	/etc/pam.d/passwd
 	/etc/pam.d/pwauth
+	/etc/ldap.conf
+	/etc/ldap.secret
 	/etc/security/pam_abl.conf
     ))
 {
@@ -523,6 +525,8 @@
 $event = "ldap-update";
 
 templates2events("/home/e-smith/ssl.pem/pem", $event);
+templates2events("/etc/ldap.conf", $event);
+templates2events("/etc/ldap.secret", $event);
 
 #--------------------------------------------------
 # actions for email-update event:
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/10ssl mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/10ssl
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/10ssl	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/10ssl	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1,2 @@
+ssl no
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/20pam_password mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/20pam_password
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/20pam_password	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/20pam_password	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1,5 @@
+pam_password exop
+pam_filter	objectclass=posixAccount
+pam_login_attribute	uid
+pam_member_attribute	memberuid
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/30host mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/30host
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/30host	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/30host	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1 @@
+host localhost
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40base mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40base
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40base	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40base	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1,3 @@
+{
+    $OUT .= "base " . esmith::util::ldapBase ($DomainName);
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_group mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_group
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_group	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_group	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1,5 @@
+{
+    $OUT .= "nss_base_group ou=Groups,";
+    $OUT .= esmith::util::ldapBase ($DomainName);
+    $OUT .= "?one\n";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_passwd mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_passwd
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_passwd	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_passwd	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1,10 @@
+{
+    $OUT .= "nss_base_passwd ou=Users,";
+    $OUT .= esmith::util::ldapBase ($DomainName);
+    $OUT .= '?one';
+    $OUT .= "\n";
+    $OUT .= "nss_base_passwd ou=Computers,";
+    $OUT .= esmith::util::ldapBase ($DomainName);
+    $OUT .= '?one';
+    $OUT .= "\n";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_shadow mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_shadow
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_shadow	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/40nss_base_shadow	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1,10 @@
+{
+    $OUT .= "nss_base_shadow ou=Users,";
+    $OUT .= esmith::util::ldapBase ($DomainName);
+    $OUT .= '?one';
+    $OUT .= "\n";
+    $OUT .= "nss_base_shadow ou=Computers,";
+    $OUT .= esmith::util::ldapBase ($DomainName);
+    $OUT .= '?one';
+    $OUT .= "\n";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/45rootbinddn mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/45rootbinddn
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/45rootbinddn	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/45rootbinddn	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1,3 @@
+{
+    $OUT .= "rootbinddn cn=root," . esmith::util::ldapBase ($DomainName);
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/55bind_policy mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/55bind_policy
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/55bind_policy	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/55bind_policy	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1,4 @@
+{
+# Allow read /etc/{passwd,groups,shadow} files when ldap is down.
+}
+bind_policy soft
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/60ignore_initgroups mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/60ignore_initgroups
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/60ignore_initgroups	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.conf/60ignore_initgroups	2010-09-23 16:26:33.000000000 +0200
@@ -0,0 +1 @@
+nss_initgroups_ignoreusers root,ldap,named,avahi,haldaemon,dbus,radvd,tomcat,radiusd,news,mailman,nscd,gdm
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.secret/all mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.secret/all
--- e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.secret/all	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/etc/ldap.secret/all	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1 @@
+{ esmith::util::LdapPassword (); }
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/ldap.secret mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/ldap.secret
--- e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/ldap.secret	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/etc/ldap.secret	2010-09-23 16:20:03.000000000 +0200
@@ -0,0 +1 @@
+PERMS=0600
