diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/home/e-smith/ssl.crt mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/home/e-smith/ssl.crt
--- e-smith-base-4.18.0/root/etc/e-smith/templates/home/e-smith/ssl.crt	2007-01-19 14:33:22.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/home/e-smith/ssl.crt	2007-03-07 11:59:18.000000000 -0700
@@ -5,12 +5,13 @@
     my $here = getcwd;
 
     my $FQDN = "$SystemName.$DomainName";
+    my $commonName = &$cropLeft($modSSL{CommonName} || $FQDN, 64);
     my $crt = "/home/e-smith/ssl.crt/$FQDN.crt";
     my $key = "/home/e-smith/ssl.key/$FQDN.key";
     my $defaultCity = $ldap{defaultCity};
     my $defaultCompany = $ldap{defaultCompany};
     my $defaultDepartment = $ldap{defaultDepartment};
-    my $email = "admin\@$FQDN";
+    my $email = "admin\@$DomainName";
 
     if ( -f $crt )
     {
@@ -24,7 +25,7 @@
             $expected_issuer .= '/L=' . ($defaultCity ? $defaultCity : 'Newbury');
             $expected_issuer .= '/O=' . ($defaultCompany ? $defaultCompany : 'My Company Ltd');
             $expected_issuer .= "/OU=$defaultDepartment" if $defaultDepartment;
-            $expected_issuer .= "/CN=$FQDN" .
+            $expected_issuer .= "/CN=$commonName" .
                                   "/emailAddress=$email";
             my $issuer = `openssl x509 -issuer -noout -in $crt`;
             chomp $issuer;
@@ -73,7 +74,6 @@
 	    $defaultCity = &$cropLeft($defaultCity, 128); 
 	    $defaultCompany = &$cropLeft($defaultCompany, 64); 
 	    $defaultDepartment = &$cropLeft($defaultDepartment, 64); 
-	    $FQDN = &$cropLeft($FQDN, 64);
 	    $email = &$cropLeft($email, 40);
 
 	    foreach (
@@ -82,7 +82,7 @@
 		    "$defaultCity",
 		    "$defaultCompany",
 		    "$defaultDepartment",
-		    "$FQDN",
+		    "$commonName",
 		    "$email"
 		    )
 	    {
