diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.1/root/etc/e-smith/web/panels/password/cgi-bin/userpassword mezzanine_patched_e-smith-base-4.18.1/root/etc/e-smith/web/panels/password/cgi-bin/userpassword
--- e-smith-base-4.18.1/root/etc/e-smith/web/panels/password/cgi-bin/userpassword	2008-08-07 20:35:22.000000000 +0200
+++ mezzanine_patched_e-smith-base-4.18.1/root/etc/e-smith/web/panels/password/cgi-bin/userpassword	2008-08-07 20:34:59.000000000 +0200
@@ -103,7 +103,7 @@
 	my $pass1 = $fm->{cgi}->param('pass');
     	unless ($pass1 eq $pass2) {
     		$fm->{cgi}->param( -name => 'wherenext', -value => 'Password' );
-       	 	return "PASSWORD_NO_MATCH";
+       	 	return "PASSWORD_VERIFY_ERROR";
     	}
         return "OK";
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.1/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm mezzanine_patched_e-smith-base-4.18.1/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm
--- e-smith-base-4.18.1/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm	2008-08-07 20:35:22.000000000 +0200
+++ mezzanine_patched_e-smith-base-4.18.1/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm	2008-08-07 20:33:44.000000000 +0200
@@ -605,7 +605,7 @@
     my $pass1 = $self->{cgi}->param('password1');
     unless ($pass1 eq $pass2) {
         $self->{cgi}->param( -name => 'wherenext', -value => 'Password' );
-        return "PASSWORD_NO_MATCH";
+        return "PASSWORD_VERIFY_ERROR";
     }
     return "OK";
 }
@@ -1140,7 +1140,7 @@
     my $pass1 = $self->{cgi}->param('pass');
     unless ($pass1 eq $pass2) {
         $self->{cgi}->param( -name => 'wherenext', -value => 'Password' );
-        return "SYSTEM_PASSWORD_NO_MATCH";
+        return "SYSTEM_PASSWORD_VERIFY_ERROR";
     }
     return "OK";
 }
