diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/events/actions/conf-modules mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/events/actions/conf-modules
--- e-smith-base-4.18.0/root/etc/e-smith/events/actions/conf-modules	2005-11-20 21:28:05.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/events/actions/conf-modules	2007-04-05 12:35:52.000000000 -0600
@@ -32,33 +32,7 @@
 while (defined (my $file = readdir(BOOT)))
 {
     next unless $file =~ /System.map-(.*)/;
-    my $version = $1;
-    open(MODULES, "</etc/modprobe.conf") or
-	die "Could not read /etc/modprobe.conf: $!";
-
-    my $pid = open(DEPMOD, "|-");
-    $SIG{ALRM} = sub { die "whoops, depmod pipe broke" };
-
-    if ($pid)
-    {  # parent
-	while (<MODULES>)
-	{
-	    s/\$\(\/bin\/uname -r\)/$version/;
-	    print DEPMOD;
-	}
-	close(DEPMOD) || warn "depmod exited $?";
-	close(MODULES);
-    }
-    else
-    {     # child
-	exec("/sbin/depmod",
-		    "-C", "/dev/stdin",
-		    "-a",
-		    "-F", "/boot/System.map-$version",
-		    "$version")
-	    or die "can't exec depmod: $!";
-	# NOTREACHED
-    }
+    exec("/sbin/depmod", "-a", "-F", "/boot/System.map-$1", "$1")
 }
 closedir(BOOT) or die("Can't close /boot: $!\n");
 
