diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.1/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/useraccounts mezzanine_patched_e-smith-base-4.18.1/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/useraccounts
--- e-smith-base-4.18.1/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/useraccounts	2008-08-07 19:09:33.000000000 +0200
+++ mezzanine_patched_e-smith-base-4.18.1/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/useraccounts	2008-08-07 19:07:36.000000000 +0200
@@ -163,10 +163,6 @@
         <trans>Enter the new password in the fields below</trans>
     </entry>
     <entry>
-        <base>RESET_PASSWORD_TITLE</base>
-        <trans>Reset user password</trans>
-    </entry>
-    <entry>
         <base>ERR_OCCURRED_MODIFYING_PASSWORD</base>
         <trans>An error occurred while updating the password</trans>
     </entry>
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.1/root/etc/e-smith/web/functions/useraccounts mezzanine_patched_e-smith-base-4.18.1/root/etc/e-smith/web/functions/useraccounts
--- e-smith-base-4.18.1/root/etc/e-smith/web/functions/useraccounts	2008-08-07 19:09:33.000000000 +0200
+++ mezzanine_patched_e-smith-base-4.18.1/root/etc/e-smith/web/functions/useraccounts	2008-08-07 19:06:01.000000000 +0200
@@ -257,7 +257,7 @@
     <page name="CheckMaxUsersUnlock" post-event="checkMaxUsers('ResetPassword')">
     </page>
     <page name="ResetPassword" pre-event="turn_off_buttons()" post-event="reset_password()">
-        <title>PASSWORD_RESET_TITLE</title>
+        <title>RESET_PASSWORD_TITLE</title>
         <subroutine src="print_page_description('reset')" />
         <field type="password" id="password1" validation="nonblank, check_password">
             <label>PASSWORD_NEW</label>
