diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/12tty2 mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/12tty2
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/12tty2	2005-11-20 21:28:07.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/12tty2	2007-05-18 00:47:09.000000000 -0600
@@ -1 +1 @@
-2:23457:respawn:/sbin/mingetty tty2
+{ -d "/proc/xen" && ! -f "/proc/xen/xsd_kva" ? "#" : "" }2:23457:respawn:/sbin/mingetty tty2
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/13tty3 mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/13tty3
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/13tty3	2005-11-20 21:28:07.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/13tty3	2007-05-18 00:47:13.000000000 -0600
@@ -1 +1 @@
-3:23457:respawn:/sbin/mingetty tty3
+{ -d "/proc/xen" && ! -f "/proc/xen/xsd_kva" ? "#" : "" }3:23457:respawn:/sbin/mingetty tty3
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/70tty1 mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/70tty1
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/70tty1	2005-11-20 21:28:07.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/inittab/70tty1	2007-05-18 00:49:37.000000000 -0600
@@ -1,7 +1,11 @@
 
 # Run gettys in standard runlevels
 {
-    if ($ConsoleMode && ($ConsoleMode ne 'auto'))
+    if ( -d "/proc/xen" && ! -f "/proc/xen/xsd_kva" )
+    {
+	"co:123457:respawn:/sbin/agetty xvc0 9600 vt100-nav";
+    }
+    elsif ($ConsoleMode && ($ConsoleMode ne 'auto'))
     {
 	"1:123457:respawn:/sbin/mingetty tty1";
     }
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/securetty/xenU-console mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/securetty/xenU-console
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/securetty/xenU-console	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/securetty/xenU-console	2007-05-18 01:17:36.000000000 -0600
@@ -0,0 +1,6 @@
+{
+    $OUT = '';
+
+    return unless -d "/proc/xen" && ! -f "/proc/xen/xsd_kva";
+    $OUT .= "xvc0";
+}
