diff -up e-smith-base-5.2.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm.fix-local-nic-string-in-console e-smith-base-5.2.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm
--- e-smith-base-5.2.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm.fix-local-nic-string-in-console	2010-06-08 21:40:27.000000000 +0200
+++ e-smith-base-5.2.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm	2010-06-08 21:41:30.000000000 +0200
@@ -116,7 +116,7 @@ sub ethernetSelect($$)
     #--------------------------------------------------------
     # These are just to ensure that xgettext knows about the
     # interface types.
-    gettext("internal");
+    gettext("local");
     gettext("external");
     #--------------------------------------------------------
 
