diff -up e-smith-base-5.0.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm.FixLocale_External e-smith-base-5.0.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm
--- e-smith-base-5.0.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm.FixLocale_External	2009-11-23 18:04:52.000000000 +0000
+++ e-smith-base-5.0.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm	2009-11-23 18:32:00.000000000 +0000
@@ -99,6 +99,13 @@ sub ethernetSelect($$)
         }
     }
 
+    #--------------------------------------------------------
+    # These are just to ensure that xgettext knows about the
+    # interface types.
+    gettext("internal");
+    gettext("external");
+    #--------------------------------------------------------
+
     push( @args,
           ++$item . ".",
           sprintf(gettext("Manually select driver for %s ethernet adapter"),
