diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/elinks.conf/10no_utf8 mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/elinks.conf/10no_utf8
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/elinks.conf/10no_utf8	2006-02-16 20:41:51.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/elinks.conf/10no_utf8	2007-03-07 12:08:01.000000000 -0700
@@ -1 +1,3 @@
 set terminal.linux.utf_8_io = 0
+set terminal.xterm.utf_8_io = 0
+set terminal.xterm.m11_hack = 1
