diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/events/actions/init-passwords mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/events/actions/init-passwords
--- e-smith-base-4.18.0/root/etc/e-smith/events/actions/init-passwords	2005-11-20 21:28:05.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/events/actions/init-passwords	2007-12-20 12:42:27.000000000 -0700
@@ -29,9 +29,9 @@
 # Lock passwords for administrative accounts.
 #------------------------------------------------------------
 
-system("/usr/bin/passwd", "-l", "root") and
+system("/usr/sbin/usermod", "-L", "root") and
     warn("Could not lock root account");
-system("/usr/bin/passwd", "-l", "admin") and
+system("/usr/sbin/usermod", "-L", "admin") and
     warn("Could not lock admin account");
 system("/usr/bin/smbpasswd", "-a", "-d", "-s", "admin") and
     warn("Could not lock admin account");
