diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/createlinks mezzanine_patched_e-smith-base-5.2.0/createlinks
--- e-smith-base-5.2.0/createlinks	2010-10-11 19:15:47.000000000 +0200
+++ mezzanine_patched_e-smith-base-5.2.0/createlinks	2010-10-11 19:10:06.000000000 +0200
@@ -15,8 +15,8 @@
 }
 
 foreach (qw(
-	/var/lib/dhcp/dhclient-eth0.conf
-	/var/lib/dhcp/dhclient-eth1.conf
+	/var/lib/dhclient/dhclient-eth0.conf
+	/var/lib/dhclient/dhclient-eth1.conf
 	/var/service/wan/pppoe.pppd.conf
 	/var/service/wan/run.pppoe.conf
 	/etc/ppp/ip-down.local
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/00interface mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/00interface
--- e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/00interface	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/00interface	2006-03-14 17:20:43.000000000 +0100
@@ -0,0 +1,4 @@
+{
+    die "Need to pass THIS_DEVICE in MORE_DATA\n" unless (defined $THIS_DEVICE);
+    $OUT = "interface \"$THIS_DEVICE\" \{";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/50hostname mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/50hostname
--- e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/50hostname	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/50hostname	2007-01-27 02:56:31.000000000 +0100
@@ -0,0 +1,7 @@
+{
+    return unless defined $ExternalInterface;
+    return unless $ExternalInterface{Name} eq $THIS_DEVICE;
+    return unless $ExternalInterface{Configuration} eq "DHCPHostname";
+    $OUT .= "    send host-name \"$DialupUserAccount\";\n";
+    $OUT .= "    send dhcp-client-identifier \"$DialupUserAccount\";";
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/template-end mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/template-end
--- e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/template-end	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhclient/dhclient.conf/template-end	2006-03-14 17:20:43.000000000 +0100
@@ -0,0 +1,3 @@
+{
+    $OUT = '\}';
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/00interface mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/00interface
--- e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/00interface	2006-03-14 17:20:43.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/00interface	1970-01-01 01:00:00.000000000 +0100
@@ -1,4 +0,0 @@
-{
-    die "Need to pass THIS_DEVICE in MORE_DATA\n" unless (defined $THIS_DEVICE);
-    $OUT = "interface \"$THIS_DEVICE\" \{";
-}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/50hostname mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/50hostname
--- e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/50hostname	2007-01-27 02:56:31.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/50hostname	1970-01-01 01:00:00.000000000 +0100
@@ -1,7 +0,0 @@
-{
-    return unless defined $ExternalInterface;
-    return unless $ExternalInterface{Name} eq $THIS_DEVICE;
-    return unless $ExternalInterface{Configuration} eq "DHCPHostname";
-    $OUT .= "    send host-name \"$DialupUserAccount\";\n";
-    $OUT .= "    send dhcp-client-identifier \"$DialupUserAccount\";";
-}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/template-end mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/template-end
--- e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/template-end	2006-03-14 17:20:43.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/template-end	1970-01-01 01:00:00.000000000 +0100
@@ -1,3 +0,0 @@
-{
-    $OUT = '\}';
-}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhclient/dhclient-eth0.conf mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhclient/dhclient-eth0.conf
--- e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhclient/dhclient-eth0.conf	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhclient/dhclient-eth0.conf	2006-03-14 17:20:43.000000000 +0100
@@ -0,0 +1,3 @@
+TEMPLATE_PATH="/var/lib/dhcp/dhclient.conf"
+OUTPUT_FILENAME="/var/lib/dhcp/dhclient-eth0.conf"
+MORE_DATA={ THIS_DEVICE => "eth0" }
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhclient/dhclient-eth1.conf mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhclient/dhclient-eth1.conf
--- e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhclient/dhclient-eth1.conf	1970-01-01 01:00:00.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhclient/dhclient-eth1.conf	2006-03-14 17:20:43.000000000 +0100
@@ -0,0 +1,3 @@
+TEMPLATE_PATH="/var/lib/dhcp/dhclient.conf"
+OUTPUT_FILENAME="/var/lib/dhcp/dhclient-eth1.conf"
+MORE_DATA={ THIS_DEVICE => "eth1" }
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhcp/dhclient-eth0.conf mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhcp/dhclient-eth0.conf
--- e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhcp/dhclient-eth0.conf	2006-03-14 17:20:43.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhcp/dhclient-eth0.conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,3 +0,0 @@
-TEMPLATE_PATH="/var/lib/dhcp/dhclient.conf"
-OUTPUT_FILENAME="/var/lib/dhcp/dhclient-eth0.conf"
-MORE_DATA={ THIS_DEVICE => "eth0" }
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhcp/dhclient-eth1.conf mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhcp/dhclient-eth1.conf
--- e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhcp/dhclient-eth1.conf	2006-03-14 17:20:43.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/etc/e-smith/templates.metadata/var/lib/dhcp/dhclient-eth1.conf	1970-01-01 01:00:00.000000000 +0100
@@ -1,3 +0,0 @@
-TEMPLATE_PATH="/var/lib/dhcp/dhclient.conf"
-OUTPUT_FILENAME="/var/lib/dhcp/dhclient-eth1.conf"
-MORE_DATA={ THIS_DEVICE => "eth1" }
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-5.2.0/root/var/service/wan/run.dhclient mezzanine_patched_e-smith-base-5.2.0/root/var/service/wan/run.dhclient
--- e-smith-base-5.2.0/root/var/service/wan/run.dhclient	2007-01-27 02:56:31.000000000 +0100
+++ mezzanine_patched_e-smith-base-5.2.0/root/var/service/wan/run.dhclient	2010-10-11 19:15:26.000000000 +0200
@@ -21,8 +21,8 @@
 
 . ./dhclient.config
 
-configfile=/var/lib/dhcp/dhclient-$interface.conf
-leasefile=/var/lib/dhcp/dhclient-$interface.leases
+configfile=/var/lib/dhclient/dhclient-$interface.conf
+leasefile=/var/lib/dhclient/dhclient-$interface.leases
 
 export PEERDNS=no
 exec /sbin/dhclient -d \
