diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm
--- e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm	2008-03-10 20:39:11.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm	2008-03-10 20:38:48.000000000 -0600
@@ -891,7 +891,7 @@
                 "3.", "www.dyndns.com" . " - " . gettext("commercial service"),
                 "4.", "www.dyndns.org" . " - " . gettext("free service"),
                 "5.", "www.tzo.com"    . " - " . gettext("commercial service"),
-                "6.", "custom",        . " - " . gettext("alternate DynDNS service"),
+                "6.", gettext("custom DynDNS service"),
                );
 
     ($rc, $choice) = $console->menu_page
