diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/ExternalInterface/10MTU mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/ExternalInterface/10MTU
--- e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/ExternalInterface/10MTU	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/ExternalInterface/10MTU	2008-01-09 20:15:54.000000000 +1100
@@ -0,0 +1 @@
+1400
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/InternalInterface/10MTU mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/InternalInterface/10MTU
--- e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/InternalInterface/10MTU	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/InternalInterface/10MTU	2008-01-09 20:15:54.000000000 +1100
@@ -0,0 +1 @@
+1400
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/sysconfig/network-scripts/ifcfg-ethX/10MTU mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/sysconfig/network-scripts/ifcfg-ethX/10MTU
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/sysconfig/network-scripts/ifcfg-ethX/10MTU	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/sysconfig/network-scripts/ifcfg-ethX/10MTU	2008-01-09 20:10:51.000000000 +1100
@@ -0,0 +1,17 @@
+{
+    my $mtu = undef;
+    $OUT = '';
+
+    if ($is_internal)
+    {
+        $mtu = $InternalInterface{MTU};
+    }
+    if ($is_external)
+    {
+        $mtu = $ExternalInterface{MTU};
+    }
+    if ($mtu)
+    {
+        $OUT .= "MTU=$mtu";
+    }
+}
