diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.1/root/usr/lib/perl5/site_perl/esmith/console/system_password.pm mezzanine_patched_e-smith-base-4.18.1/root/usr/lib/perl5/site_perl/esmith/console/system_password.pm
--- e-smith-base-4.18.1/root/usr/lib/perl5/site_perl/esmith/console/system_password.pm	2008-03-26 10:49:00.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.18.1/root/usr/lib/perl5/site_perl/esmith/console/system_password.pm	2008-03-26 10:56:00.000000000 -0600
@@ -67,19 +67,19 @@
     # Cracklib strings.
     # Note the extra space here and in the gettext call below. This
     # allows the French localization to properly generate qu'il
-    gettext(" it is based on your username");
-    gettext(" it is based upon your password entry");
-    gettext(" it is derived from your password entry");
-    gettext(" it is derivable from your password entry");
-    gettext(" it is too short");
-    gettext(" it is all whitespace");
-    gettext(" it is too simplistic/systematic");
-    gettext(" it is based on a dictionary word");
-    gettext(" it is based on a (reversed) dictionary word");
-    gettext(" it does not contain numbers");
-    gettext(" it does not contain uppercase characters");
-    gettext(" it does not contain lowercase characters");
-    gettext(" it does not contain special characters");
+    gettext("it is based on your username");
+    gettext("it is based upon your password entry");
+    gettext("it is derived from your password entry");
+    gettext("it is derivable from your password entry");
+    gettext("it is too short");
+    gettext("it is all whitespace");
+    gettext("it is too simplistic/systematic");
+    gettext("it is based on a dictionary word");
+    gettext("it is based on a (reversed) dictionary word");
+    gettext("it does not contain numbers");
+    gettext("it does not contain uppercase characters");
+    gettext("it does not contain lowercase characters");
+    gettext("it does not contain special characters");
     #--------------------------------------------------------
 
     my $strength = $db->get_prop("passwordstrength", "Admin");
