diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm
--- e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm	2007-01-26 20:56:31.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/configure.pm	2007-02-22 18:08:48.000000000 -0500
@@ -2079,20 +2079,11 @@
     $db->set_prop("bootstrap-console", "ForceSave", "no");
     $db->set_prop("bootstrap-console", "Run", "no");
 
-    # XXX-FIXME
-    # We call whiptail directly (rather than through the screen function)
-    # as we don't want the --clear parameter passed by screen().
-    # Appropriate magic can clean this up.
-
-    system(
-           "/usr/bin/whiptail",
-           "--backtitle", $console->backtitle,
-           "--title", gettext("Activating configuration settings"),
-           "--infobox", gettext("Please stand by while your configuration settings are activated ..."),
-           8, esmith::console::SCREEN_COLUMNS,
+    $console->infobox(
+           title => gettext("Activating configuration settings"),
+           text => gettext("Please stand by while your configuration settings are activated ..."),
           );
 
-
     if ($bootstrapConsole eq "yes")
     {
         system("/sbin/e-smith/signal-event", "bootstrap-console-save");
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/save_config.pm mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/save_config.pm
--- e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/save_config.pm	2007-02-22 18:12:44.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/save_config.pm	2007-02-22 18:08:08.000000000 -0500
@@ -22,17 +22,9 @@
     $db->set_prop("bootstrap-console", "ForceSave", "no");
     $db->set_prop("bootstrap-console", "Run", "no");
 
-    # XXX-FIXME
-    # We call whiptail directly (rather than through the screen function)
-    # as we don't want the --clear parameter passed by screen().
-    # Appropriate magic can clean this up.
-
-    system(
-           "/usr/bin/whiptail",
-           "--backtitle", $console->backtitle,
-           "--title", gettext("Activating configuration settings"),
-           "--infobox", gettext("Please standby while your configuration settings are activated ..."),
-           8, esmith::console::SCREEN_COLUMNS,
+    $console->infobox(
+           title => gettext("Activating configuration settings"),
+           text => gettext("Please standby while your configuration settings are activated ..."),
           );
 
     system("/sbin/e-smith/signal-event", 'bootstrap-console-save');
