diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/usr/share/hal/fdi/95userpolicy/usb-rev35-drive.fdi mezzanine_patched_e-smith-base-4.18.0/root/usr/share/hal/fdi/95userpolicy/usb-rev35-drive.fdi
--- e-smith-base-4.18.0/root/usr/share/hal/fdi/95userpolicy/usb-rev35-drive.fdi	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/usr/share/hal/fdi/95userpolicy/usb-rev35-drive.fdi	2008-01-07 21:09:34.000000000 -0700
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="ISO-8859-1"?> <!-- -*- SGML -*- -->
+<deviceinfo version="0.2">
+  <device>
+      <match key="storage.bus" string="usb">
+        <match key="storage.vendor" string="Iomega">
+          <match key="storage.model" string="RRD">
+            <merge key="storage.policy.desired_mount_point" type="string">usbdisk</merge>
+          </match>
+        </match>
+      </match>
+  </device>
+</deviceinfo>
