diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/logrotate.d/httpd/00notused mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/logrotate.d/httpd/00notused
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/logrotate.d/httpd/00notused	2007-09-11 15:35:35.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/logrotate.d/httpd/00notused	1969-12-31 17:00:00.000000000 -0700
@@ -1,4 +0,0 @@
-#
-# We do not use logrotate to rotate apache controlled logfiles,
-# we reconfigure apache instead. No action required here.
-#
