diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/perform_backup.pm mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/perform_backup.pm
--- e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/perform_backup.pm	2007-04-11 04:55:53.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/perform_backup.pm	2007-04-11 04:55:41.000000000 -0600
@@ -27,8 +27,6 @@
 
 sub order
 {
-    return -1 if (defined $ARGV [0]);
-
     return $_[0]->{order};
 }
 
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/quitConsole.pm mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/quitConsole.pm
--- e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/quitConsole.pm	2007-01-26 18:56:31.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/quitConsole.pm	2007-04-11 04:55:17.000000000 -0600
@@ -21,8 +21,6 @@
 
 sub order
 {
-    return -1 if (defined $ARGV [0]);
-
     return $_[0]->{order};
 }
 
