diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/sbin/e-smith/console mezzanine_patched_e-smith-base-4.15.4/root/sbin/e-smith/console
--- e-smith-base-4.15.4/root/sbin/e-smith/console	2005-06-07 09:32:56.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.15.4/root/sbin/e-smith/console	2005-07-17 12:48:23.737077600 -0600
@@ -2636,7 +2636,7 @@
          text  =>
          gettext("This option will start a text-mode browser to access the server manager from this console.  Normally you would access the server manager from a web browser at the following url:") .
          "\n\n" .
-         "http://$conf{'SystemName'}/server-manager/" .
+         "https://$conf{'SystemName'}/server-manager/" .
          "\n\n" .
          gettext("You should only proceed if you are comfortable using a text-mode browser.  Note that you will be prompted for the administrator password in order to access the server manager.") .
          "\n\n" .
@@ -2667,7 +2667,7 @@
                    "-restrictions=all",
                    "-rlogin",
                    "-telnet",
-                   "http://localhost/server-manager/noframes"
+                   "http://localhost:980/server-manager/noframes"
                   );
         }
     }
