diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/web/functions/groups mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/web/functions/groups
--- e-smith-base-4.15.4/root/etc/e-smith/web/functions/groups	2005-03-10 13:57:23.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/web/functions/groups	2005-07-27 00:37:59.866719272 -0600
@@ -7,7 +7,7 @@
 # navigation  : 2000 2200
 #----------------------------------------------------------------------
 #----------------------------------------------------------------------
-# copyright (C) 2002 Mitel Networks Corporation
+# copyright (C) 2002-2005 Mitel Networks Corporation
 # 
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -114,9 +114,8 @@
         post-event="create_group()">
         <title>CREATE_TITLE</title>
         <description>GROUP_NAMING</description>
-        <subroutine src="warn_if_max_groups()"/>
         <field type="text" id="groupName" 
-            validation="validate_groups_available, validate_group, validate_group_length, validate_group_naming_conflict">
+            validation="validate_group, validate_group_length, validate_group_naming_conflict">
             <label>GROUP_NAME</label>
         </field>
         <field type="text" id="groupDesc" validation="validate_description">
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/groups.pm mezzanine_patched_e-smith-base-4.15.4/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/groups.pm
--- e-smith-base-4.15.4/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/groups.pm	2005-05-12 15:44:29.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.15.4/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/groups.pm	2005-07-27 00:37:30.809136696 -0600
@@ -20,14 +20,12 @@
 our @EXPORT = qw(
 
   show_initial
-  warn_if_max_groups
   genUsers
   create_group
   modify_group
   delete_group
   validate_is_group
   validate_group_naming_conflict
-  validate_groups_available
   validate_group
   validate_group_length
   getNextFreeID
@@ -41,9 +39,6 @@
   get_cgi_param
 );
 
-use vars qw($MAX_GROUPS);
-$MAX_GROUPS = 28;
-
 our $accounts = esmith::AccountsDB->open() || die "Couldn't open accounts";
 our $db = esmith::ConfigDB->open || die "Couldn't open config db";
 
@@ -162,15 +157,13 @@
 
     my $params = $fm->build_cgi_params();
 
-    warn_if_max_groups($fm);
-
     my $numGroups = $accounts->groups;
-    if ( $numGroups != $MAX_GROUPS ) {
-        print $q->Tr($q->td(
+
+    print $q->Tr($q->td(
 	  "<p><a class=\"button-like\" href=\"groups?$params&wherenext=CreateGroup\">"
           . $fm->localise("CLICK_TO_CREATE")
           . "</a></p>"));
-    }
+
     if ( $numGroups == 0 ) {
         print $q->Tr($q->td(
 	    '<p><b>' . $fm->localise("NO_GROUPS") . '</p></b>'));
@@ -347,25 +340,6 @@
 	    });
 }
 
-
-=head2 validate_groups_available
-
-    returns OK if new groups can be created.
-    returns 'MAX_GROUPS_IN_USE' if we don't have any free slots
-
-=cut
-
-sub validate_groups_available {
-    my $numGroups = $accounts->groups();
-    if ( $numGroups >= $MAX_GROUPS ) {
-        return ('MAX_GROUPS_IN_USE');
-    }
-    else {
-        return ('OK');
-    }
-}
-
-
 =head2 validate_group FM groupname
 
 Returns OK if the group name contains only valid characters
@@ -555,22 +529,6 @@
     return $fm->props_to_query_string( \%props );
 }
 
-
-=head2 warn_if_max_groups
-
-Prints a warning if all user groups are in use
-
-=cut
-
-sub warn_if_max_groups() {
-    my $fm = shift;
-    unless ( validate_groups_available() eq 'OK' ) {
-        print $fm->localise("MAX_GROUPS_IN_USE");
-    }
-
-    return;
-}
-
 =head2 genUsers MEMBERS
 
 Takes a comma delimited list of users and returns a string of 
