diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/sbin/e-smith/console-menu-items/manageRAID.pl mezzanine_patched_e-smith-base-4.18.0/root/sbin/e-smith/console-menu-items/manageRAID.pl
--- e-smith-base-4.18.0/root/sbin/e-smith/console-menu-items/manageRAID.pl	2007-01-26 18:56:31.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/sbin/e-smith/console-menu-items/manageRAID.pl	2007-03-07 11:35:17.000000000 -0700
@@ -144,7 +144,7 @@
 	"\n"
         ;
 
-    ($rc, $choice) = $console->yesno_page(title => $title, text => $text);
+    ($rc, $choice) = $console->yesno_page(title => $title, text => $text, defaultno => 1);
     return unless ($rc == 0);
 
     unless (system(@cmd) == 0)
