diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.0/root/etc/e-smith/web/functions/useraccounts mezzanine_patched_e-smith-base-4.17.0/root/etc/e-smith/web/functions/useraccounts
--- e-smith-base-4.17.0/root/etc/e-smith/web/functions/useraccounts	2006-02-16 20:41:51.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.0/root/etc/e-smith/web/functions/useraccounts	2007-01-14 20:07:25.000000000 -0700
@@ -230,6 +230,18 @@
         <field type="text" id="LastName" validation="nonblank">
             <label>LASTNAME</label>
         </field>
+        <field type="select" id="EmailForward" options="'local' =>
+            'DELIVER_EMAIL_LOCALLY', 'forward' => 'FORWARD_EMAIL',
+            'both' => 'DELIVER_AND_FORWARD'" validation="nonblank" value='local'
+            display="display_email_forwarding()"
+            >
+            <label>EMAIL_DELIVERY</label>
+        </field>
+        <field type="text" id="ForwardAddress" validation="emailforward()"
+            display="display_email_forwarding()"
+            >
+            <label>FORWARDING_ADDRESS</label>
+        </field>
         <field type="select" id="VPNClientAccess" options="'yes' => 'YES',
             'no' => 'NO'" validation="nonblank" value="get_pptp_value()">
             <label>VPN_CLIENT_ACCESS</label>
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm mezzanine_patched_e-smith-base-4.17.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm
--- e-smith-base-4.17.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm	2006-02-16 20:41:51.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm	2007-01-14 20:10:16.000000000 -0700
@@ -696,6 +696,8 @@
     my %newProperties = (
         'FirstName'      => $self->{cgi}->param('FirstName'),
         'LastName'       => $self->{cgi}->param('LastName'),
+        'EmailForward'   => $self->{cgi}->param('EmailForward'),
+        'ForwardAddress' => $self->{cgi}->param('ForwardAddress'),
         'VPNClientAccess'=> $self->{cgi}->param('VPNClientAccess'),
     );
 
