diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.16.0/root/var/service/raidmonitor/run mezzanine_patched_e-smith-base-4.16.0/root/var/service/raidmonitor/run
--- e-smith-base-4.16.0/root/var/service/raidmonitor/run	2006-02-16 20:41:51.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.16.0/root/var/service/raidmonitor/run	2006-12-23 12:32:48.000000000 -0700
@@ -1,9 +1,11 @@
 #! /bin/sh
 
 exec 2>&1
-if grep -q '(hd1)' /boot/grub/device.map
+count=$(sed -n '/[0-9]\s\+[^ ]\+[^0-9]$/p;/[0-9]\s\+\(rd\|ida\|cciss\|i2o\)\/[^ ]\+[^0-9p][0-9]\+$/p' /proc/partitions | wc -l)
+if [ $count -ge 2 ]
 then
     exec /sbin/mdadm --monitor --scan --program /sbin/e-smith/mdevent
+else
+    echo "Only $count drive(s) in this system - disabling raid monitoring"
+    runsvctrl d .
 fi
-echo This appears to be a single disk system - disabling raid monitoring
-runsvctrl d .
