diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/events/actions/group-create-unix mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/events/actions/group-create-unix
--- e-smith-base-4.15.4/root/etc/e-smith/events/actions/group-create-unix	2005-04-01 09:52:47.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/events/actions/group-create-unix	2005-07-19 14:37:53.005892616 -0600
@@ -55,7 +55,7 @@
 {
     use esmith::lockfile;
 
-    $lock = esmith::lockfile::LockFileOrWait("/home/e-smith/accounts");
+    $lock = esmith::lockfile::LockFileOrWait("/home/e-smith/db/accounts");
     $gid = $accounts->get_next_uid;
     $group->set_prop('Gid', $gid);
     unless ($group->prop('Uid'))
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/events/actions/user-create-unix mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/events/actions/user-create-unix
--- e-smith-base-4.15.4/root/etc/e-smith/events/actions/user-create-unix	2005-04-01 12:53:56.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/events/actions/user-create-unix	2005-07-19 14:37:53.009892008 -0600
@@ -52,7 +52,7 @@
 {
     use esmith::lockfile;
 
-    $lock = esmith::lockfile::LockFileOrWait("/home/e-smith/accounts");
+    $lock = esmith::lockfile::LockFileOrWait("/home/e-smith/db/accounts");
     $uid = $accounts->get_next_uid;
     $acct->set_prop('Uid', $uid);
 }
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/tests/10e-smith-base/configuration.conf mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/tests/10e-smith-base/configuration.conf
--- e-smith-base-4.15.4/root/etc/e-smith/tests/10e-smith-base/configuration.conf	2005-03-10 13:57:23.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/tests/10e-smith-base/configuration.conf	2005-07-19 14:37:52.771928184 -0600
@@ -33,7 +33,7 @@
 LocalNetmask=255.255.255.0
 MinUid=5000
 PasswordSet=yes
-PreviousConfiguration=/home/e-smith/configuration.previous
+PreviousConfiguration=/home/e-smith/db/configuration.previous
 SMTPSmartHost=
 SambaDomainMaster=no
 SambaServerName=pretz
