diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/createlinks mezzanine_patched_e-smith-base-4.15.8/createlinks
--- e-smith-base-4.15.8/createlinks	2006-01-29 21:06:10.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/createlinks	2006-01-29 21:02:50.000000000 -0500
@@ -167,9 +167,6 @@
 	email-update
 	logrotate
 	));
-templates2events("/etc/statusreport",  qw(
-	bootstrap-console-save
-	));
 
 # conf-routes
 templates2events("/etc/sysconfig/network-scripts/route-eth0", qw(
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/templates/etc/crontab/20statusreport mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/etc/crontab/20statusreport
--- e-smith-base-4.15.8/root/etc/e-smith/templates/etc/crontab/20statusreport	2005-11-20 23:28:06.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/etc/crontab/20statusreport	1969-12-31 19:00:00.000000000 -0500
@@ -1,19 +0,0 @@
-{
-    my $result = "";
-
-    # Add cron rules for status reporting
-
-    if ($StatusReports eq 'on')
-    {
-        # pick a random minute (0-59) and hour (0-23)
-
-	srand (time);
-	my $minute = int (rand (60));
-	my $hour   = int (rand (24));
-
-        $result .= "$minute\t$hour\t*\t*\t*\tpublic\t/etc/statusreport\n";
-    }
-
-    $result;
-}
-
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/templates/etc/statusreport/00startup mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/etc/statusreport/00startup
--- e-smith-base-4.15.8/root/etc/e-smith/templates/etc/statusreport/00startup	2005-11-20 23:28:08.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/etc/statusreport/00startup	1969-12-31 19:00:00.000000000 -0500
@@ -1,47 +0,0 @@
-package esmith;
-
-use strict;
-
-open(UPTIME, "/proc/uptime") or die "Can't open /proc/uptime";
-my ($uptime) = split(/\s/, <UPTIME>);
-close UPTIME;
-
-#------------------------------------------------------------
-# Work out the current release
-#------------------------------------------------------------
-my $release = "(unknown version)";
-
-if ( open(RPM, "/bin/rpm -q e-smith-release|") )
-\{
-    $release = <RPM>;
-    close RPM;
-
-    chomp($release);
-    $release =~ s/e-smith-release-//;
-\}
-
-my $cmd = "/usr/bin/wget --output-document=- --tries=1"
-{
-    $OUT = '';
-    if ($SquidParent && $SquidParent ne '')
-    {
-	my $port = $SquidParentPort || "8080";
-
-	$OUT .= " . \" --proxy=on -e ";
-	$OUT .= "http_proxy=$SquidParent:";
-	$OUT .= "$port\"\n";
-    }
-} . " \"http://servertest.e-smith.com/report.phtml?"
- . "state=serverStatusAddCommit&"
- . "esmithVersion=$release&"
- . "name={ "$SystemName.$DomainName" }&"
- . "uptime=$uptime\""
- . " >/dev/null 2>&1";
-
-my $status = system ($cmd);
-
-my $log = ( $status == 0 ) ? "OK" : "Failed - status $status";
-
-system ( '/usr/bin/logger', '-it', 'statusreport', $log );
-
-exit 0;
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/init_config mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/init_config
--- e-smith-base-4.15.8/root/sbin/e-smith/init_config	2005-11-20 23:28:11.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/init_config	2006-01-29 21:05:51.000000000 -0500
@@ -90,7 +90,6 @@
 sub ethernetSelect($$);
 sub isValidIP($);
 sub rebootShutdown();
-sub testInternet();
 
 my $release = esmith::util::determineRelease();
 my $release_string = "Mitel Networks server ${release}";
@@ -2744,94 +2743,6 @@
     sleep(600);
 }
 
-#------------------------------------------------------------
-# TEST:
-#------------------------------------------------------------
-sub testInternet()
-{
-    use LWP;
-    use HTTP::Request::Common;
-
-    ($rc, $choice) = $console->yesno_page
-        (
-         title => gettext("Test Internet access"),
-         text  =>
-         gettext("After your Internet connection is operational and your server has been connected and configured, this test will verify that your server can communicate with the Internet.") .
-         "\n\n" .
-         gettext("As part of this test, a summary of your system configuration will be sent to Mitel Networks. User account names and passwords will not be sent. We appreciate this opportunity to be notified of successful server installations, and will not release this information to anyone outside of Mitel Networks.") .
-         "\n\n" .
-         gettext("Do you wish to run the test?"),
-        );
-
-    return unless ($rc == 0);
-
-    # get system parameters (filter out accounts and passwords)
-
-    my @results = ();
-    foreach my $key (keys %conf)
-    {
-        unless (($key =~ /Account$/) || ($key =~ /Password$/))
-        {
-            push @results, $key;
-        }
-    }
-
-    use constant SERVERTEST => 'http://servertest.e-smith.com/report.phtml';
-
-    my $ua = LWP::UserAgent->new;
-
-    # add proxy arguments if appropriate
-    if ($conf{'SquidParent'} && ($conf{'SquidParent'} ne ''))
-    {
-        my $parent = $conf{'SquidParent'};
-        my $port = $conf{'SquidParentPort'} || "3128";
-        $ua->proxy(['http', 'ftp'], "http://$parent:$port/");
-    }
-    my $request = POST(SERVERTEST,
-                       Content => [
-                                   state => 'serverTestAddCommitVersion41',
-                                   esmithVersion => $release,
-                                   map { $_ => $conf{$_} } @results,
-                                  ],
-                      );
-
-    print '-' x 75 . "\n";
-    print gettext("Attempting to test Internet connection.") . "\n";
-    print gettext("If this test takes too long, please halt it by typing Ctrl-C.") . "\n";
-    print '-' x 75 . "\n";
-
-    my $response = $ua->request($request);
-
-    if ($response->is_success)
-    {
-        ($rc, $choice) = $console->message_page
-            (
-             title => gettext("Internet connection successful"),
-             text  =>
-             gettext("The test was successful! Your server was able to communicate with Mitel Networks."),
-            );
-    }
-    else
-    {
-        warn gettext("Response to Internet test was"), ": ",
-            $response->message,
-                "\n";
-
-        ($rc, $choice) = $console->message_page
-            (
-             title => "Internet connection failed",
-             text  =>
-             gettext("The test failed. Your server was unable to establish contact with Mitel Networks via the Internet.") .
-             "\n\n" .
-             gettext("Please check that your server is correctly configured. A reboot may be required if certain settings are modified and the configuration process will advise you if this is required.") .
-             "\n\n" .
-             gettext("You might also want to check with your Internet provider to make sure that your Internet connection is working properly.") .
-             "\n\n" .
-             gettext("It is also possible (although unlikely) that there is a technical problem at the Mitel Networks site. In this case, you should proceed with your installation and run this test again later."),
-            );
-    }
-}
-
 sub build_menu
 {
     my @items = ();
