diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/sysconfig/Registration mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/sysconfig/Registration
--- e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/sysconfig/Registration	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/sysconfig/Registration	2005-07-19 14:37:52.403984120 -0600
@@ -0,0 +1 @@
+none
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/db/configuration/migrate/10SystemID mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/migrate/10SystemID
--- e-smith-base-4.15.4/root/etc/e-smith/db/configuration/migrate/10SystemID	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/migrate/10SystemID	2005-07-19 14:37:52.405983816 -0600
@@ -0,0 +1,7 @@
+{
+    use Data::UUID; 
+   
+    my $sysconfig = $DB->get('sysconfig') || $DB->new_record('sysconfig', { type => 'configuration'});
+
+    $sysconfig->prop('SystemID') || $sysconfig->set_prop("SystemID", new Data::UUID->create_str);
+}
