diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/sbin/e-smith/add_mirror mezzanine_patched_e-smith-base-4.15.4/root/sbin/e-smith/add_mirror
--- e-smith-base-4.15.4/root/sbin/e-smith/add_mirror	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.4/root/sbin/e-smith/add_mirror	2005-10-13 17:30:43.158027450 +1000
@@ -0,0 +1,84 @@
+#!/bin/sh
+#----------------------------------------------------------------------
+# add_mirror: Add second half of mirror to one-way RAID
+#----------------------------------------------------------------------
+# Copyright (C) 2005 Gordon Rowell <gordonr@gormand.com.au>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
+#----------------------------------------------------------------------
+
+OLD=$1
+NEW=$2
+
+usage()
+{
+    echo $@
+    echo "Usage: add_mirror existing_disk new_disk"
+    exit 1
+}
+
+[ "$OLD" = "$NEW" ] && usage "$OLD and $NEW are the same disk"
+
+for disk in $OLD $NEW
+do
+    [ -b /dev/$disk ] || usage "/dev/$disk is not a block special device"
+done
+
+case $OLD in
+    hd*) IDE=/proc/ide
+
+        [ $(cat $IDE/$OLD/media) = 'disk' ] || usage "$OLD is not a disk"
+
+        [ $(cat $IDE/$NEW/media) = 'disk' ] || usage "$NEW is not a disk"
+
+        [ $(cat $IDE/$NEW/capacity) = $(cat $IDE/$OLD/capacity) ] || \
+            usage "$OLD and $NEW are different size disks"
+        # XXX - FIXME - should we check geometry here?
+        
+        ;;
+
+        *)  # XXX - FIXME - we should do the same tests for SCSI disks ;;
+esac
+
+part_check='No partitions found'
+
+NEW_parts=$(sfdisk -l /dev/$NEW 2>&1 | grep "$part_check")
+
+[ "$NEW_parts" = "$part_check" ] || \
+    usage "$NEW already has partitions configured"
+
+sfdisk -d /dev/$OLD >/var/tmp/oldparts.$OLD
+
+sfdisk --no-reread --force /dev/$NEW < /var/tmp/oldparts.$OLD
+
+for part in 1 2 3
+do
+    grep -s "^md${part} "  /proc/mdstat || continue
+
+    mdadm --add /dev/md$part /dev/${NEW}${part}
+done
+
+# We should be able to do grub-install --recheck /dev/md1, but not in
+# this version of grub-install
+# We need to rebuild the device.map file since $NEW didn't exist
+# at install time.
+grub-install --recheck /dev/$OLD
+# grub-install /dev/$NEW
+
+grub --batch <<HERE
+install --stage2=/boot/grub/stage2 (hd0,0)/grub/stage1 (hd0) (hd0,0)/grub/stage2 p /grub/grub.conf
+install --stage2=/boot/grub/stage2 (hd1,0)/grub/stage1 (hd1) (hd1,0)/grub/stage2 p /grub/grub.conf
+quit
+HERE
