diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/console-menu-items/manageRAID.pl mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console-menu-items/manageRAID.pl
--- e-smith-base-4.15.8/root/sbin/e-smith/console-menu-items/manageRAID.pl	2006-02-01 04:52:08.400246308 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console-menu-items/manageRAID.pl	2006-02-01 04:49:05.397086801 +1100
@@ -132,7 +132,7 @@
     {
         $text = gettext("The command @cmd failed.") .
                 "\n\n" .
-                gettext("Manual intervention is required.");
+                gettext("This configuration is not yet fully supported in these screens.");
 
         my ($rc, $choice) = $console->message_page(title => $title, text => $text);
         return;
