diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.16.0/root/sbin/e-smith/console mezzanine_patched_e-smith-base-4.16.0/root/sbin/e-smith/console
--- e-smith-base-4.16.0/root/sbin/e-smith/console	2006-03-15 03:20:43.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.16.0/root/sbin/e-smith/console	2006-04-05 14:12:32.482304509 +1000
@@ -310,6 +310,7 @@
     gettext(" it is too simplistic/systematic");
     gettext(" it is based on a dictionary word");
     gettext(" it is based on a (reversed) dictionary word");
+    gettext(" it is too long");
     #--------------------------------------------------------
 
     my $reason;
@@ -318,6 +319,10 @@
     {
 	$reason = "it is too short";
     }
+    elsif (length($_) > 14)
+    {
+	$reason = "it is too long";
+    }
     elsif (!(/\d/ and /[A-Z]/ and /[a-z]/ and /\W/))
     {
 	$reason = "it is too simplistic/systematic";
