diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/templates/etc/sysconfig/network-scripts/route-ethX/10localroutes mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/etc/sysconfig/network-scripts/route-ethX/10localroutes
--- e-smith-base-4.15.8/root/etc/e-smith/templates/etc/sysconfig/network-scripts/route-ethX/10localroutes	2005-11-21 15:28:09.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/etc/sysconfig/network-scripts/route-ethX/10localroutes	2005-11-30 17:58:54.705408842 +1100
@@ -37,6 +37,7 @@
 	$OUT .= "ADDRESS$index=$net\n";
 	$OUT .= "NETMASK$index=$netmask\n";
 	$OUT .= "GATEWAY$index=$router\n";
+	$index++;
     }
 }
 
