diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.1/root/usr/lib/perl5/site_perl/esmith/console/backup_running.pm mezzanine_patched_e-smith-base-4.17.1/root/usr/lib/perl5/site_perl/esmith/console/backup_running.pm
--- e-smith-base-4.17.1/root/usr/lib/perl5/site_perl/esmith/console/backup_running.pm	2007-01-18 20:01:57.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.1/root/usr/lib/perl5/site_perl/esmith/console/backup_running.pm	2007-01-19 08:35:34.000000000 -0700
@@ -19,6 +19,8 @@
     # check whether a backup in process and incomplete
     #-------------------------------------------------------------
     my $restore_db = esmith::ConfigDB->open_ro("/etc/e-smith/restore");
+    return unless $restore_db;
+
     my $restore_state = $restore_db->get_prop('restore', 'state')
 	|| 'idle';
 
