diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/console mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console
--- e-smith-base-4.15.8/root/sbin/e-smith/console	2005-12-16 15:22:53.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console	2005-12-16 15:21:57.000000000 -0500
@@ -318,7 +318,17 @@
     gettext(" it is based on a (reversed) dictionary word");
     #--------------------------------------------------------
 
-    my $reason = fascist_check($choice1, '/usr/lib/cracklib_dict');
+    my $reason;
+    $_ = $choice1;
+    if (length($_) < 7)
+    {
+	$reason = "it is too short";
+    }
+    elsif (!(/\d/ and /[A-Z]/ and /[a-z]/ and /\W/))
+    {
+	$reason = "it is too simplistic/systematic";
+    }
+    $reason ||= fascist_check($choice1, '/usr/lib/cracklib_dict');
     # Untaint return data from cracklib, so we can use it later. We
     # trust the library, so we accept anything.
     $reason =~ /(.+)/; $reason = $1;
