diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.9/root/sbin/e-smith/console-menu-items/manageRAID.pl mezzanine_patched_e-smith-base-4.15.9/root/sbin/e-smith/console-menu-items/manageRAID.pl
--- e-smith-base-4.15.9/root/sbin/e-smith/console-menu-items/manageRAID.pl	2006-02-17 14:41:51.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.9/root/sbin/e-smith/console-menu-items/manageRAID.pl	2006-03-07 17:52:01.297501471 +1100
@@ -71,12 +71,19 @@
     }
 
     my @unclean = ();
+    my @recovering = ();
     my %used_disks = ();
 
     for my $dev (keys %devices)
     {
 	$used_disks{$_}++ for (@{$devices{$dev}{UsedDisks}});
 
+        if ($devices{$dev}{State} =~ /recovering/)
+        {
+            push @recovering, "$dev => " . $devices{$dev}{State};
+            next;
+        }
+
         next if ($devices{$dev}{State} eq "clean");
 
         push @unclean, "$dev => " . $devices{$dev}{State};
@@ -86,6 +93,15 @@
 
     warn "unclean: @unclean\n" if DEBUG_MANAGE_RAID;
 
+    warn "recovering: @recovering\n" if DEBUG_MANAGE_RAID;
+
+    if (scalar @recovering)
+    {
+        $text .= gettext("A RAID resynchronization is in progress.");
+        my ($rc, $choice) = $console->message_page(title => $title, text => $text);
+        return;
+    }
+
     unless (scalar @unclean)
     {
         $text .= gettext("All RAID devices are in clean state");
