diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/console mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console
--- e-smith-base-4.15.8/root/sbin/e-smith/console	2006-02-17 11:15:06.205300575 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console	2006-02-17 09:59:47.000000000 +1100
@@ -2820,14 +2820,15 @@
     my @args =
         (
          gettext("Reboot"),     gettext("Reboot this server"),
+	 gettext("Reconfigure"),   gettext("Reconfigure this server"),
          gettext("Shutdown"),   gettext("Shutdown this server"),
         );
 
     ($rc, $choice) = $console->menu_page
         (
-         title => gettext("Reboot or shutdown this server"),
+         title => gettext("Reboot, Reconfigure or shutdown this server"),
          text  =>
-         gettext("Please select whether you wish to reboot or shutdown. The process will start as soon as you make your selection.") .
+         gettext("Please select whether you wish to reboot, reconfigure or shutdown. The process will start as soon as you make your selection.") .
          "\n\n" .
          gettext("If you have an older computer without power management, the shutdown process will perform a clean halt of all system services, but will not actually power off your computer. In this case, wait for the power down message and then shut off the power manually.") .
          "\n\n" .
@@ -2844,11 +2845,17 @@
         system("/usr/bin/tput", "clear");
         system("/sbin/e-smith/signal-event", "halt");
     }
-    else
+    elsif ($choice eq gettext('Reboot'))
     {
         system("/usr/bin/tput", "clear");
         system("/sbin/e-smith/signal-event", "reboot");
     }
+    elsif ($choice eq gettext('Reconfigure'))
+    {
+        system("/usr/bin/tput", "clear");
+	system("/sbin/e-smith/signal-event", "post-upgrade");
+        system("/sbin/e-smith/signal-event", "reboot");
+    }
 
     # A bit of a hack to avoid the console restarting before the
     # reboot takes effect.
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/console-menu-items/reboot.pl mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console-menu-items/reboot.pl
--- e-smith-base-4.15.8/root/sbin/e-smith/console-menu-items/reboot.pl	2005-11-21 15:28:10.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console-menu-items/reboot.pl	2006-02-17 10:59:15.000000000 +1100
@@ -4,7 +4,7 @@
 {
     my $class = shift;
     my $self = {
-		    name => "Reboot or shut down this server",
+		    name => "Reboot, Reconfigure or shut down this server",
 		    order => 40,
 		};
     bless $self, $class;
