diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/modSSL/CipherSuite mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/modSSL/CipherSuite
--- e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/modSSL/CipherSuite	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/modSSL/CipherSuite	2005-12-15 11:22:50.617656331 +1100
@@ -0,0 +1 @@
+ALL:!aNULL:!ADH:!eNULL:!LOW:!EXP:RC4+RSA:+HIGH:+MEDIUM
