diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.0/root/etc/e-smith/db/configuration/migrate/10SystemLocalNetwork mezzanine_patched_e-smith-base-4.17.0/root/etc/e-smith/db/configuration/migrate/10SystemLocalNetwork
--- e-smith-base-4.17.0/root/etc/e-smith/db/configuration/migrate/10SystemLocalNetwork	2005-11-20 23:28:05.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.17.0/root/etc/e-smith/db/configuration/migrate/10SystemLocalNetwork	2006-08-13 06:37:52.000000000 -0400
@@ -2,13 +2,13 @@
     use esmith::util;
     use esmith::NetworksDB;
 
-    my $LocalIP = $DB->get('LocalIP');
-    return unless defined $LocalIP;    # Nothing to migrate yet
-    $LocalIP = $LocalIP->value;
-
-    my $LocalNetmask = $DB->get('LocalNetmask');
-    return unless defined $LocalNetmask;
-    $LocalNetmask = $LocalNetmask->value;
+    my $InternalInterface = $DB->get('InternalInterface');
+    return unless $InternalInterface;;    # Nothing to migrate yet
+
+    my $LocalIP = $InternalInterface->prop('IPAddress');
+    my $LocalNetmask = $InternalInterface->prop('Netmask');
+
+    my $server_only = ($DB->get_value('SystemMode') eq 'serveronly');
 
     my $ndb = esmith::NetworksDB->open
       || esmith::NetworksDB->create;
@@ -17,13 +17,21 @@
     my ($localnet) = $ndb->get_all_by_prop( SystemLocalNetwork => 'yes' );
     my ($local_network) =
       esmith::util::computeNetworkAndBroadcast( $LocalIP, $LocalNetmask );
-    if ( defined $localnet && $localnet->key ne $local_network )
+    if (defined $localnet)
     {
-
-        # We need to delete the old record
-        $localnet->delete;
-        $localnet = undef;
+        if ($localnet->key ne $local_network)
+        {
+            # We need to delete the old record
+            $localnet->delete;
+            $localnet = undef;
+        }
+        if ($server_only)
+        {
+            $localnet->delete_prop('SystemLocalNetwork');
+        }
     }
+    return if $server_only; # serveronly mode does not get a 'local network' by default
+
     if ( !defined $localnet )
     {
         # We need to convert an existing local network to system network
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/localnetworks mezzanine_patched_e-smith-base-4.17.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/localnetworks
--- e-smith-base-4.17.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/localnetworks	2005-11-20 23:28:05.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.17.0/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/localnetworks	2006-08-13 06:49:40.000000000 -0400
@@ -102,14 +102,6 @@
         </trans>
     </entry>
     <entry>
-        <base>NETWORK_ALREADY_ADDED</base>
-        <trans>
-            Error: network {$network} (derived from network
-            {$networkAddress} and subnet mask {$networkMask}) 
-            has already been added. Did not add new network.
-        </trans>
-    </entry>
-    <entry>
         <base>ERROR_CREATING_NETWORK</base>
         <trans>
             Error occurred while creating network.
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/localnetworks.pm mezzanine_patched_e-smith-base-4.17.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/localnetworks.pm
--- e-smith-base-4.17.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/localnetworks.pm	2006-03-14 11:20:43.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.17.0/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/localnetworks.pm	2006-08-13 06:49:26.000000000 -0400
@@ -339,15 +339,9 @@
     my ( $network, $broadcast ) =
       esmith::util::computeNetworkAndBroadcast( $networkAddress, $networkMask );
 
-    if ( $network eq $localNetwork )
-    {
-        $fm->error('NETWORK_ALREADY_LOCAL');
-        return;
-    }
-
     if ( $network_db->get($network) )
     {
-        $fm->error('NETWORK_ALREADY_ADDED');
+        $fm->error('NETWORK_ALREADY_LOCAL');
         return;
     }
 
