diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/console mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console
--- e-smith-base-4.15.8/root/sbin/e-smith/console	2005-11-21 15:28:10.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console	2005-11-30 18:26:01.205416118 +1100
@@ -2384,7 +2384,7 @@
            "--backtitle",
 	   "$release_string Copyright 1999-2005 Mitel Networks Corporation",
            "--title", gettext("Activating configuration settings"),
-           "--infobox", gettext("Please standby while your configuration settings are activated ..."),
+           "--infobox", gettext("Please stand by while your configuration settings are activated ..."),
            8, SCREEN_COLUMNS,
           );
 
