diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/sysconfig/PreviousSystemMode mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/sysconfig/PreviousSystemMode
--- e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/sysconfig/PreviousSystemMode	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/sysconfig/PreviousSystemMode	2005-12-14 12:00:37.719835143 +1100
@@ -0,0 +1 @@
+unknown
