diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.9/root/sbin/e-smith/add_mirror mezzanine_patched_e-smith-base-4.15.9/root/sbin/e-smith/add_mirror
--- e-smith-base-4.15.9/root/sbin/e-smith/add_mirror	2006-02-17 14:41:51.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.9/root/sbin/e-smith/add_mirror	2006-03-07 17:05:52.225406889 +1100
@@ -19,6 +19,9 @@
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
 #----------------------------------------------------------------------
 
+PATH=$PATH:/sbin
+export PATH
+
 usage()
 {
     echo $@
@@ -43,6 +46,9 @@
 
 [ "$OLD" = "$NEW" ] && usage "$OLD and $NEW are the same disk"
 
+metadevices=$(mdadm --detail --scan | grep "/dev/${NEW}1")
+[ -z "$metadevices" ] || usage "$NEW is already in use"
+
 for disk in $OLD $NEW
 do
     [ -b /dev/$disk ] || usage "/dev/$disk is not a block special device"
