diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.16.0/root/sbin/e-smith/console-menu-items/manageRAID.pl mezzanine_patched_e-smith-base-4.16.0/root/sbin/e-smith/console-menu-items/manageRAID.pl
--- e-smith-base-4.16.0/root/sbin/e-smith/console-menu-items/manageRAID.pl	2006-03-14 11:20:43.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.16.0/root/sbin/e-smith/console-menu-items/manageRAID.pl	2006-04-19 14:34:57.000000000 -0400
@@ -159,7 +159,10 @@
     $text = $disk_status . 
         "\n\n" . 
         gettext("There is an unused disk drive in your system. Do you want to add it to the existing drive to make a RAID pair?") . 
-        "\n\n";
+	"\n\n" .
+	gettext("WARNING: ALL DATA ON THE NEW DISK WILL BE DESTROYED!") .
+	"\n" .
+        ;
 
     my ($rc, $choice) = $console->yesno_page(title => $title, text => $text);
     return unless ($rc == 0);
