diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.1/createlinks mezzanine_patched_e-smith-base-4.18.1/createlinks
--- e-smith-base-4.18.1/createlinks	2008-03-26 10:49:00.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.18.1/createlinks	2008-03-26 14:56:57.000000000 -0600
@@ -431,7 +431,7 @@
 
 $event = "user-delete";
 
-event_link("user-delete-groups-and-pseudonyms", $event, "10");
+event_link("user-delete-groups-and-pseudonyms", $event, "02");
 event_link("user-delete-unix", $event, "15");
 event_link("initialize-default-databases", $event, "23");
 event_link("count-active-user-accounts", $event, "25");
