diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/access mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/access
--- e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/access	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/access	2005-07-30 17:16:52.646959000 -0600
@@ -0,0 +1 @@
+local
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/TCPPort mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/TCPPort
--- e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/TCPPort	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/TCPPort	2005-07-30 17:16:52.647959000 -0600
@@ -0,0 +1 @@
+980
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager
--- e-smith-base-4.15.4/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager	2005-06-07 09:32:56.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/templates/etc/httpd/admin-conf/httpd.conf/20Manager	2005-07-30 17:16:52.648959000 -0600
@@ -1,5 +1,5 @@
 {
-    $OUT .= "Listen 127.0.0.1:980\n";
+    $OUT .= "Listen 127.0.0.1:${'httpd-admin'}{TCPPort}\n";
 
     $OUT .= <<HERE;
 
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/templates/etc/services/20http-admin mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/templates/etc/services/20http-admin
--- e-smith-base-4.15.4/root/etc/e-smith/templates/etc/services/20http-admin	2002-02-15 12:31:21.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/templates/etc/services/20http-admin	2005-07-30 17:16:52.649959000 -0600
@@ -1 +1 @@
-http-admin	980/tcp		# admin HTTP server
+http-admin	{ ${'httpd-admin'}{TCPPort} }/tcp		# admin HTTP server
