diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/access mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/access
--- e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/access	2005-08-02 18:41:39.829527408 -0600
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/httpd-admin/access	2005-08-02 18:37:34.213668776 -0600
@@ -1 +1 @@
-local
+localhost
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/sbin/e-smith/console mezzanine_patched_e-smith-base-4.15.4/root/sbin/e-smith/console
--- e-smith-base-4.15.4/root/sbin/e-smith/console	2005-08-02 18:41:39.837518192 -0600
+++ mezzanine_patched_e-smith-base-4.15.4/root/sbin/e-smith/console	2005-08-02 18:37:28.922473160 -0600
@@ -2667,7 +2667,7 @@
                    "-restrictions=all",
                    "-rlogin",
                    "-telnet",
-                   "https://localhost/server-manager/noframes"
+                   "http://localhost:980/server-manager/noframes"
                   );
         }
     }
