diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.16.0/root/etc/e-smith/templates/home/e-smith/ssl.crt mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/templates/home/e-smith/ssl.crt
--- e-smith-base-4.16.0/root/etc/e-smith/templates/home/e-smith/ssl.crt	2005-11-20 23:28:09.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/templates/home/e-smith/ssl.crt	2006-05-28 19:33:15.000000000 -0400
@@ -6,17 +6,34 @@
     my $FQDN = "$SystemName.$DomainName";
     my $crt = "/home/e-smith/ssl.crt/$FQDN.crt";
     my $key = "/home/e-smith/ssl.key/$FQDN.key";
+    my $defaultCity = $ldap{defaultCity};
+    my $defaultCompany = $ldap{defaultCompany};
+    my $defaultDepartment = $ldap{defaultDepartment};
+    my $email = "admin\@$FQDN";
 
     if ( -f $crt && (-M $crt < KEYLIFEINDAYS))
     {
-	# Old key file is still good. Read it out - processTemplate will work
-	# out that it hasn't changed, and leave the old one in place
-	open(C, "$crt") or die "Couldn't open crt file: $!";
-	my @crt = <C>;
-	chomp @crt;
-	$OUT = join "\n", @crt;
-	close(C);
-	return;
+        my $expected_issuer = '/C=--' .
+                              '/ST=----' .
+                              "/L=$defaultCity" .
+                              "/O=$defaultCompany" .
+                              "/OU=$defaultDepartment" .
+                              "/CN=$FQDN" .
+                              "/emailAddress=$email";
+        my $issuer = `openssl x509 -issuer -noout -in $crt`;
+        chomp $issuer;
+        $issuer =~ s/^issuer= //;
+        if ($issuer eq $expected_issuer)
+	{
+	    # Old key file is still good. Read it out - processTemplate will work
+	    # out that it hasn't changed, and leave the old one in place
+	    open(C, "$crt") or die "Couldn't open crt file: $!";
+	    my @crt = <C>;
+	    chomp @crt;
+	    $OUT = join "\n", @crt;
+	    close(C);
+	    return;
+	}
     }
     # go to somewhere private and safe where we can run programs
     # as root
@@ -34,10 +51,6 @@
 	if ($pid)
 	{
 	    # parent
-	    my $defaultCity = $ldap{defaultCity};
-	    my $defaultCompany = $ldap{defaultCompany};
-	    my $defaultDepartment = $ldap{defaultDepartment};
-	    my $email = "admin\@$FQDN";
 
 	    # anonymous sub to crop a variable to a specified size
 	    my $cropLeft = sub { 
