diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/directive mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/directive
--- e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/directive	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/directive	2007-12-24 01:54:54.000000000 -0700
@@ -0,0 +1 @@
+-M diminishing
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/email mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/email
--- e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/email	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/email	2007-12-24 01:53:24.000000000 -0700
@@ -0,0 +1 @@
+-m admin
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/status mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/status
--- e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/status	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/status	2007-12-24 01:52:42.000000000 -0700
@@ -0,0 +1 @@
+disabled
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/type mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/type
--- e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/type	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/db/configuration/defaults/smartd/type	2005-11-20 21:28:05.000000000 -0700
@@ -0,0 +1 @@
+service
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/etc/smartd.conf/10conf mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/smartd.conf/10conf
--- e-smith-base-4.18.0/root/etc/e-smith/templates/etc/smartd.conf/10conf	1969-12-31 17:00:00.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/etc/smartd.conf/10conf	2007-12-24 02:03:56.000000000 -0700
@@ -0,0 +1,8 @@
+{
+# /etc/smartd.conf
+
+# Sample configuration file for smartd.  See man 5 smartd.conf.
+# Home page is: http://smartmontools.sourceforge.net
+
+"DEVICESCAN -a $smartd{email} $smartd{directive} -I 190 -I 194";
+}
