diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/createlinks mezzanine_patched_e-smith-base-4.15.8/createlinks
--- e-smith-base-4.15.8/createlinks	2005-12-14 15:30:08.675630096 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/createlinks	2005-12-14 15:29:46.926414215 +1100
@@ -583,6 +583,7 @@
 
 my %service2order =
 (
+    microcode_ctl => "00",
     # Start syslog up before network!
     syslog   	=> "05",
     #klogd   	=> "06",
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/microcode_ctl/status mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/microcode_ctl/status
--- e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/microcode_ctl/status	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/microcode_ctl/status	2005-12-14 15:28:44.386295493 +1100
@@ -0,0 +1 @@
+enabled
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/microcode_ctl/type mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/microcode_ctl/type
--- e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/microcode_ctl/type	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/defaults/microcode_ctl/type	2005-12-14 15:28:47.939677244 +1100
@@ -0,0 +1 @@
+service
