diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/events/actions/user-modify-unix mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/events/actions/user-modify-unix
--- e-smith-base-4.15.8/root/etc/e-smith/events/actions/user-modify-unix	2005-11-20 23:28:05.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/events/actions/user-modify-unix	2006-02-08 18:17:17.000000000 -0500
@@ -62,7 +62,7 @@
     #------------------------------------------------------------
     unless ($shell eq $new_shell)
     {
-	system("/usr/sbin/usermod", '-s', "$shell", $userName) == 0
+	system("/usr/sbin/usermod", '-s', "$new_shell", $userName) == 0
 	    or die "Failed to modify shell of account $userName.\n";
     }
 
