diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/createlinks mezzanine_patched_e-smith-base-4.15.8/createlinks
--- e-smith-base-4.15.8/createlinks	2006-02-12 19:12:27.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/createlinks	2006-02-12 19:10:04.000000000 -0500
@@ -588,9 +588,9 @@
     syslog   	=> "05",
     #klogd   	=> "06",
     cpuspeed   	=> "06",
+    'bootstrap-console'	=> "10",
     raidmonitor => "15",
     apmd	=> "26",
-    'bootstrap-console'	=> "35",
     # Set up the packet filter rules before bringing up networks.
     # But, after the bootstrap-console has had a chance to configure masq
     masq 	=> "36",
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/rc.d/init.d/bootstrap-console mezzanine_patched_e-smith-base-4.15.8/root/etc/rc.d/init.d/bootstrap-console
--- e-smith-base-4.15.8/root/etc/rc.d/init.d/bootstrap-console	2005-11-20 23:28:10.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/rc.d/init.d/bootstrap-console	2006-02-12 19:11:22.000000000 -0500
@@ -1,7 +1,7 @@
 #!/bin/sh
 
 #----------------------------------------------------------------------
-# copyright (C) 1999, 2000 e-smith, inc.
+# copyright (C) 1999-2006 Mitel Networks Corporation
 #
 # This program is free software; you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -17,8 +17,6 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307  USA
 #
-# Technical support for this program is available from e-smith, inc.
-# Please visit our web site www.e-smith.com for details.
 #----------------------------------------------------------------------
 
 #----------------------------------------------------------------------
@@ -44,14 +42,10 @@
 		yes) /sbin/e-smith/console console
 		     ;;
 
-		no)  # And now we'd better restart anything which
+		no)  # And now we'd better restart syslog which
 		     # may have been configured before the changes in the
 		     # console
-		     for earlyservice in random syslog
-		     do
-		         /etc/rc.d/init.d/${earlyservice} restart
-		     done
-
+		     /etc/rc.d/init.d/syslog restart
 		     break
 		     ;;
 	    esac
