diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/passwordstrength/Ibays mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/passwordstrength/Ibays
--- e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/passwordstrength/Ibays	2005-04-13 15:14:34.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/passwordstrength/Ibays	2005-07-19 14:37:52.487971352 -0600
@@ -1 +1 @@
-normal
+strong
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/passwordstrength/Users mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/passwordstrength/Users
--- e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/passwordstrength/Users	2005-04-13 15:14:34.000000000 -0600
+++ mezzanine_patched_e-smith-base-4.15.4/root/etc/e-smith/db/configuration/defaults/passwordstrength/Users	2005-07-19 14:37:52.493970440 -0600
@@ -1 +1 @@
-normal
+strong
