diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.4/root/sbin/e-smith/console mezzanine_patched_e-smith-base-4.15.4/root/sbin/e-smith/console
--- e-smith-base-4.15.4/root/sbin/e-smith/console	2005-07-30 17:25:37.533264440 -0600
+++ mezzanine_patched_e-smith-base-4.15.4/root/sbin/e-smith/console	2005-07-30 17:25:25.227135256 -0600
@@ -65,7 +65,7 @@
 
         return undef if ($self->EXISTS($key) and $self->FETCH($key) eq $value);
 
-        if (($self->filename eq 'configuration')
+        if (($self->filename =~ m#(^|/)configuration$#)
             && ($key ne 'UnsavedChanges')) {
             $self->SUPER::STORE('UnsavedChanges', 'yes');
         }
