diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.16.0/root/etc/e-smith/events/actions/rmmod-bonding mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/events/actions/rmmod-bonding
--- e-smith-base-4.16.0/root/etc/e-smith/events/actions/rmmod-bonding	2006-03-14 11:20:43.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/events/actions/rmmod-bonding	2006-04-18 16:13:42.000000000 -0400
@@ -2,11 +2,14 @@
 
 # This is an ugly hack that removes the bonding module during
 # bootstrap-console-save. We do this because we can potentially 
-# change modprobe.conf in the event, but bonding.ko is already
-# loaded by the time we do. This means the changes don't take
+# change modprobe.conf in the event, but bonding.ko may already
+# be loaded by the time we do (by /etc/rc.sysinit).
+# This means the changes don't take
 # effect unless you reboot (ugh). Removing the module before
 # networking is started causes the module to be reloaded using
 # the new module params.
 
-/sbin/rmmod bonding
-
+if grep -q bonding /proc/modules
+then
+  /sbin/rmmod bonding
+fi
