diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/createlinks mezzanine_patched_e-smith-base-4.15.8/createlinks
--- e-smith-base-4.15.8/createlinks	2006-02-17 14:37:27.945774170 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/createlinks	2006-02-17 14:36:34.291109479 +1100
@@ -318,6 +318,8 @@
 foreach (qw(
     /root/upgrade.log
     /root/upgrade.log.syslog
+    /var/log/anaconda.log
+    /var/log/anaconda.syslog
 	))
 {
     safe_touch "root/etc/e-smith/events/post-upgrade/logfiles2timestamp/$_";
