diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/var/service/raidmonitor/run mezzanine_patched_e-smith-base-4.18.0/root/var/service/raidmonitor/run
--- e-smith-base-4.18.0/root/var/service/raidmonitor/run	2007-01-19 14:33:22.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/var/service/raidmonitor/run	2007-02-16 13:28:37.000000000 -0700
@@ -7,5 +7,5 @@
     exec /sbin/mdadm --monitor --scan --program /sbin/e-smith/mdevent
 else
     echo "Only $count drive(s) in this system - disabling raid monitoring"
-    runsvctrl d .
+    sv d .
 fi
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/var/service/wan/run.dialup mezzanine_patched_e-smith-base-4.18.0/root/var/service/wan/run.dialup
--- e-smith-base-4.18.0/root/var/service/wan/run.dialup	2007-01-26 18:56:31.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.18.0/root/var/service/wan/run.dialup	2007-02-16 13:28:22.000000000 -0700
@@ -3,7 +3,7 @@
 ISDN=$(/sbin/e-smith/config getprop isdn status)
 if [ "$ISDN" = "enabled" ]
 then
-    runsvctrl u /service/ippp
+    sv u /service/ippp
     sleep 10
     # TODO check here that ISDN device is available!!
 fi
