diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.16.0/root/etc/e-smith/events/actions/copy-anaconda-logs mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/events/actions/copy-anaconda-logs
--- e-smith-base-4.16.0/root/etc/e-smith/events/actions/copy-anaconda-logs	2006-03-15 03:20:43.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/events/actions/copy-anaconda-logs	2006-03-15 10:03:47.401718937 +1100
@@ -26,6 +26,11 @@
 
 for i in $LIST
 do
-    [ -f $i ] && cp $i $i.$NOW
+    [ -e $i ] || continue
+
+    [ -L $i ] && continue
+
+    mv $i $i.$NOW
+    ln -s $i.$NOW $i
 done
 
