diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/templates/etc/sysconfig/network/60NOZEROCONF mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/etc/sysconfig/network/60NOZEROCONF
--- e-smith-base-4.15.8/root/etc/e-smith/templates/etc/sysconfig/network/60NOZEROCONF	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/etc/sysconfig/network/60NOZEROCONF	2006-02-01 03:13:46.730070601 +1100
@@ -0,0 +1 @@
+NOZEROCONF=yes
