diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.16.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/50hostname mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/50hostname
--- e-smith-base-4.16.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/50hostname	2006-11-21 08:44:04.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/templates/var/lib/dhcp/dhclient.conf/50hostname	2006-11-21 08:43:43.000000000 -0500
@@ -1,4 +1,5 @@
 {
+    return unless $ExternalInterface;
     return unless $ExternalInterface{Name} eq $THIS_DEVICE;
     return unless $ExternalInterface{Configuration} eq "DHCPHostname";
     $OUT .= "    send host-name \"$DialupUserAccount\";\n";
