diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/system_password.pm mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/system_password.pm
--- e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/system_password.pm	2008-01-09 17:32:33.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.18.0/root/usr/lib/perl5/site_perl/esmith/console/system_password.pm	2008-01-09 17:32:18.000000000 +1100
@@ -60,35 +60,6 @@
         goto INITIAL_PASSWORD;
     }
 
-
-    ($rc, $choice2) = $console->password_page
-        (
-         title   => gettext("Choose administrator password"),
-         text    => gettext("Please type your administrator password again to verify."),
-        );
-
-    unless ($rc == 0)
-    {
-        ($rc, $choice) = $console->message_page
-            (
-             title => gettext("Administrator password not set"),
-             text  => gettext("Sorry, you must set the administrator password now."),
-            );
-
-        goto INITIAL_PASSWORD;
-    }
-
-    if ($choice1 ne $choice2)
-    {
-        ($rc, $choice) = $console->message_page
-            (
-             title => gettext("Passwords do not match"),
-             text  => gettext("The two passwords did not match"),
-            );
-
-        goto INITIAL_PASSWORD;
-    }
-
     use Crypt::Cracklib;
 
     #--------------------------------------------------------
@@ -133,6 +104,34 @@
         goto INITIAL_PASSWORD if ($rc == 0);
     }
 
+    ($rc, $choice2) = $console->password_page
+        (
+         title   => gettext("Choose administrator password"),
+         text    => gettext("Please type your administrator password again to verify."),
+        );
+
+    unless ($rc == 0)
+    {
+        ($rc, $choice) = $console->message_page
+            (
+             title => gettext("Administrator password not set"),
+             text  => gettext("Sorry, you must set the administrator password now."),
+            );
+
+        goto INITIAL_PASSWORD;
+    }
+
+    if ($choice1 ne $choice2)
+    {
+        ($rc, $choice) = $console->message_page
+            (
+             title => gettext("Passwords do not match"),
+             text  => gettext("The two passwords did not match"),
+            );
+
+        goto INITIAL_PASSWORD;
+    }
+
     #--------------------------------------------------
     # Set system password
     #--------------------------------------------------
