diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.0/root/etc/e-smith/templates/home/e-smith/ssl.crt mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/home/e-smith/ssl.crt
--- e-smith-base-4.18.0/root/etc/e-smith/templates/home/e-smith/ssl.crt	2007-10-11 18:57:07.000000000 -0400
+++ mezzanine_patched_e-smith-base-4.18.0/root/etc/e-smith/templates/home/e-smith/ssl.crt	2007-10-11 18:48:06.000000000 -0400
@@ -13,6 +13,13 @@
     my $defaultDepartment = $ldap{defaultDepartment};
     my $email = "admin\@$DomainName";
 
+    # crop fields that are too long for X509:
+    $defaultCity = substr($defaultCity, 0, 128); 
+    $defaultCompany = substr($defaultCompany, 0, 64); 
+    $defaultDepartment = substr($defaultDepartment, 0, 64); 
+    $email = substr($email, 0, 40);
+    $commonName = substr($commonName, 0, 64);
+
     if ( -f $crt )
     {
         my $expire = `openssl x509 -enddate -noout -in $crt`;
@@ -60,23 +67,6 @@
 	{
 	    # parent
 
-	    # anonymous sub to crop a variable to a specified size
-	    my $cropLeft = sub { 
-		my ($var, $len) = @_;
-		if (length($var) > $len)
-		{ 
-		    $var = substr($var, 0, $len);
-		}
-		return $var;
-	    };
-
-	    # crop fields that are too long for X509:
-	    $defaultCity = &$cropLeft($defaultCity, 128); 
-	    $defaultCompany = &$cropLeft($defaultCompany, 64); 
-	    $defaultDepartment = &$cropLeft($defaultDepartment, 64); 
-	    $email = &$cropLeft($email, 40);
-	    $commonName = &$cropLeft($commonName, 64);
-
 	    foreach (
 		    "--",
 		    "----",
