diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/console mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console
--- e-smith-base-4.15.8/root/sbin/e-smith/console	2005-12-06 05:15:37.821777274 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/console	2005-12-06 05:12:27.678860005 +1100
@@ -2086,7 +2086,7 @@
     goto SYSTEM_MODE unless ($rc == 0);
 
     if ($choice eq gettext("On")
-        || ($choice eq gettext("keep") && $DHCPServer eq "On")) {
+        || ($choice eq gettext("keep") && $DHCPServer eq gettext("On"))) {
         db_set_prop(\%conf, 'dhcpd', 'status', 'enabled');
         # Initialise Samba DomainMaster setting if it isn't already set
         # default to yes if DHCP is enabled
