diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.16.0/root/etc/e-smith/templates/home/e-smith/ssl.crt mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/templates/home/e-smith/ssl.crt
--- e-smith-base-4.16.0/root/etc/e-smith/templates/home/e-smith/ssl.crt	2006-05-28 21:13:11.000000000 -0400
+++ mezzanine_patched_e-smith-base-4.16.0/root/etc/e-smith/templates/home/e-smith/ssl.crt	2006-05-28 19:38:40.000000000 -0400
@@ -91,6 +91,7 @@
 		qw(req -new -key),
 		$key,
 		qw(-x509 -days), 
+		qw(-set_serial), time(),
 		KEYLIFEINDAYS,
 		)
 		    || die "can't exec program: $!";
