diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/accounts/00openRW mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/accounts/00openRW
--- e-smith-base-4.15.8/root/etc/e-smith/db/accounts/00openRW	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/accounts/00openRW	2006-02-05 20:26:22.000000000 -0500
@@ -0,0 +1,4 @@
+{
+    use esmith::AccountsDB;
+    $DB = esmith::AccountsDB->open;
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/domains/00openRW mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/domains/00openRW
--- e-smith-base-4.15.8/root/etc/e-smith/db/domains/00openRW	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/domains/00openRW	2006-02-05 20:26:58.000000000 -0500
@@ -0,0 +1,5 @@
+{
+    use esmith::DomainsDB;
+    $DB = esmith::DomainsDB->open;
+}
+
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/hosts/00openRW mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/hosts/00openRW
--- e-smith-base-4.15.8/root/etc/e-smith/db/hosts/00openRW	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/hosts/00openRW	2006-02-05 20:26:41.000000000 -0500
@@ -0,0 +1,4 @@
+{
+    use esmith::HostsDB;
+    $DB = esmith::HostsDB->open;
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/networks/00openRW mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/networks/00openRW
--- e-smith-base-4.15.8/root/etc/e-smith/db/networks/00openRW	1969-12-31 19:00:00.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/networks/00openRW	2006-02-05 20:31:45.000000000 -0500
@@ -0,0 +1,4 @@
+{
+    use esmith::NetworksDB;
+    $DB = esmith::NetworksDB->open;
+}
