diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/templates/usr/lib/e-smith-mdevent/mdEvent.tmpl mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/usr/lib/e-smith-mdevent/mdEvent.tmpl
--- e-smith-base-4.15.8/root/etc/e-smith/templates/usr/lib/e-smith-mdevent/mdEvent.tmpl	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/templates/usr/lib/e-smith-mdevent/mdEvent.tmpl	2006-01-17 13:09:13.000000000 +1100
@@ -0,0 +1,23 @@
+{
+    use esmith::I18N;
+    use Locale::gettext;
+
+    my $i18n = new esmith::I18N;
+    $i18n->setLocale('mdEvent.tmpl');
+
+    my $domain = $conf->get_value("DomainName") || "localhost";
+    my $systemName = $conf->get_value("SystemName") || "SME Server";
+
+    $OUT .= "To: $user\n";
+    $OUT .= "From: \"".  gettext("mdadm monitoring") . "\" <root\@${domain}>\n";
+    $OUT .= "Subject: " . gettext("$event event on") .  " ${device}:${systemName}.${domain}\n\n";
+
+    $OUT .= gettext("This is an automatically generated mail message from mdadm running on").
+        " ${systemName}.${domain}.\n\n";
+
+    $OUT .= gettext("A $event event has been detected on md device").
+        " ${device}.\n\n";
+
+    $OUT .= gettext("Device") . " $member ". gettext("is now an active member of md device") . 
+        " ${device}.\n" if $member;
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/mdevent mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/mdevent
--- e-smith-base-4.15.8/root/sbin/e-smith/mdevent	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/mdevent	2006-01-17 13:03:46.000000000 +1100
@@ -0,0 +1,45 @@
+#!/usr/bin/perl -w
+
+package esmith;
+
+use strict;
+
+use esmith::ConfigDB;
+use esmith::util;
+use esmith::db;
+
+use Text::Template;
+
+my $conf = esmith::ConfigDB->open_ro;
+
+my $event = $ARGV [0];
+my $device = $ARGV [1];
+my $member = $ARGV [2] || '';
+
+print "Event: $event, Device: $device, Member: $member\n";
+
+if ($event =~ m#^Rebuild|^Fail|^Degraded|^SpareActive#) {
+    my $domain = $conf->get_value("DomainName") || 'localhost';
+    my $user = "admin-raidreport\@$domain";
+
+    my $templates = '/etc/e-smith/templates';
+    my $source = '/usr/lib/e-smith-mdevent/mdEvent.tmpl';
+
+    -f "${templates}-custom${source}" and $templates .= "-custom";
+
+    my $t = new Text::Template(TYPE => 'FILE',
+    SOURCE => "${templates}${source}");
+
+    open(QMAIL, "|/var/qmail/bin/qmail-inject -froot\@$domain $user")
+    || die "Could not send mail via qmail-inject!\n";
+
+    print QMAIL $t->fill_in( HASH => {
+        conf    => \$conf,
+        user    => $user,
+        event   => $event,
+        device  => $device,
+        member  => $member,
+    });
+
+    close QMAIL;
+}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/var/service/raidmonitor/run mezzanine_patched_e-smith-base-4.15.8/root/var/service/raidmonitor/run
--- e-smith-base-4.15.8/root/var/service/raidmonitor/run	2005-11-21 15:28:11.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/var/service/raidmonitor/run	2006-01-17 13:02:09.000000000 +1100
@@ -3,7 +3,7 @@
 exec 2>&1
 if grep -q '(hd1)' /boot/grub/device.map
 then
-    exec /sbin/mdadm --monitor --scan --mail admin-raidreport
+    exec /sbin/mdadm --monitor --scan --program /sbin/e-smith/mdevent
 fi
 echo This appears to be a single disk system - disabling raid monitoring
 runsvctrl d .
