diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.18.1/root/etc/e-smith/web/panels/password/cgi-bin/userpassword mezzanine_patched_e-smith-base-4.18.1/root/etc/e-smith/web/panels/password/cgi-bin/userpassword
--- e-smith-base-4.18.1/root/etc/e-smith/web/panels/password/cgi-bin/userpassword	2008-08-08 12:42:22.000000000 +0200
+++ mezzanine_patched_e-smith-base-4.18.1/root/etc/e-smith/web/panels/password/cgi-bin/userpassword	2008-08-08 12:42:00.000000000 +0200
@@ -142,7 +142,7 @@
             <label>PASSWORD_NEW</label>
         </field>
         <field type="password" id="passVerify" validation="password_compare" value="">
-            <label>PASSWORD_NEW_VERIFY</label>
+            <label>PASSWORD_VERIFY_NEW</label>
         </field>
         <subroutine src="print_button('PASSWORD_CHANGE')" />
     </page>
