diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/createlinks mezzanine_patched_e-smith-base-4.17.2/createlinks
--- e-smith-base-4.17.2/createlinks	2007-01-19 16:30:40.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/createlinks	2007-01-19 16:23:00.000000000 -0700
@@ -70,17 +70,6 @@
     templates2events($_, qw(console-save bootstrap-console-save));
 }
 
-# conf-masq
-
-templates2events("/etc/rc.d/init.d/masq", qw(
-	console-save
-	bootstrap-console-save
-	network-create
-	network-delete
-	remoteaccess-update
-	email-update
-	));
-
 # conf-modules
 
 templates2events("/etc/modprobe.conf", qw(
@@ -303,7 +292,6 @@
 event_link("reset-unsavedflag", $event, "95");
 event_link("init-reload", $event, "90");
 
-safe_symlink("adjust", "root/etc/e-smith/events/$event/services2adjust/masq");
 safe_symlink("restart", "root/etc/e-smith/events/$event/services2adjust/wan");
 safe_symlink("restart", "root/etc/e-smith/events/$event/services2adjust/dhcpd");
 
@@ -365,8 +353,6 @@
 event_link("set-external-ip", $event, "15");
 event_link("update-dns", $event, "85");
 
-safe_symlink("adjust", "root/etc/e-smith/events/$event/services2adjust/masq");
-
 #--------------------------------------------------
 # actions for network-create event
 #--------------------------------------------------
@@ -375,8 +361,6 @@
 
 event_link("conf-routes", $event, "89");
 
-safe_symlink("adjust", "root/etc/e-smith/events/$event/services2adjust/masq");
-
 #--------------------------------------------------
 # actions for network-delete event
 #--------------------------------------------------
@@ -385,8 +369,6 @@
 
 event_link("conf-routes", $event, "89");
 
-safe_symlink("adjust", "root/etc/e-smith/events/$event/services2adjust/masq");
-
 #--------------------------------------------------
 # actions for post-install event
 #--------------------------------------------------
@@ -431,7 +413,6 @@
 
 $event = "remoteaccess-update";
 
-safe_symlink("adjust", "root/etc/e-smith/events/$event/services2adjust/masq");
 safe_symlink("restart", "root/etc/e-smith/events/$event/services2adjust/dhcpd");
 
 #--------------------------------------------------
@@ -552,9 +533,6 @@
     'bootstrap-console'	=> "10",
     raidmonitor => "15",
     apmd	=> "26",
-    # Set up the packet filter rules before bringing up networks.
-    # But, after the bootstrap-console has had a chance to configure masq
-    masq 	=> "36",
     network  	=> "37",
     irqbalance 	=> "39",
     crond    	=> "40",
@@ -585,7 +563,5 @@
 # Set up links to daemontools.
 safe_symlink("daemontools", "root/etc/rc.d/init.d/raidmonitor");
 
-# Set up links to daemontools.
 safe_symlink("run.dhclient", "root/var/service/wan/run.DHCPHostname");
 safe_symlink("run.dhclient", "root/var/service/wan/run.DHCPEthernetAddress");
-
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/Logging mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/Logging
--- e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/Logging	2005-11-20 21:28:04.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/Logging	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-most
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/status mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/status
--- e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/status	2005-11-20 21:28:04.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/status	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-enabled
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/Trace mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/Trace
--- e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/Trace	2005-11-20 21:28:04.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/Trace	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-disabled
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/type mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/type
--- e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/type	2005-11-20 21:28:04.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/defaults/masq/type	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-service
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/db/configuration/migrate/10EnableMasq mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/migrate/10EnableMasq
--- e-smith-base-4.17.2/root/etc/e-smith/db/configuration/migrate/10EnableMasq	2007-01-19 14:33:22.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/db/configuration/migrate/10EnableMasq	1969-12-31 17:00:00.000000000 -0700
@@ -1,5 +0,0 @@
-{
-    return if $DB->get_value('SystemMode') eq 'serveronly';
-
-    $DB->set_prop('masq', 'status', 'enabled');
-}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates/etc/identd.masq/50localaddresses mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/identd.masq/50localaddresses
--- e-smith-base-4.17.2/root/etc/e-smith/templates/etc/identd.masq/50localaddresses	2005-11-20 21:28:07.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/identd.masq/50localaddresses	1969-12-31 17:00:00.000000000 -0700
@@ -1,25 +0,0 @@
-{
-    #--------------------------------------------------
-    # Compute local IP address, netmask and network values.
-    #--------------------------------------------------
-
-    my $ipaddrBits  = esmith::util::IPquadToAddr ($LocalIP);
-    my $netmaskBits = esmith::util::IPquadToAddr ($LocalNetmask);
-    my $networkBits = $ipaddrBits & $netmaskBits;
-
-    #--------------------------------------------------
-    # Compute our hostid, and the highest hostid, limiting range
-    # to a class B at most (so we don't get a huge output file).
-    #--------------------------------------------------
-
-    my $myHostid = (~ $netmaskBits) & $ipaddrBits;
-
-    my $maxHostid = ((~ $netmaskBits) & 0xffffff) - 1;
-    $maxHostid = ($maxHostid <= 65534) ? $maxHostid : 65534;
-
-    for ($i = 1; $i <= $maxHostid; $i++)
-    {
-        my $ip = esmith::util::IPaddrToQuad ($networkBits | $i);
-    	$OUT .= "$ip\tuser$i\t\tUNIX\n";
-    }
-}
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/00start mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/00start
--- e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/00start	2005-11-20 21:28:08.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/00start	1969-12-31 17:00:00.000000000 -0700
@@ -1,5 +0,0 @@
-
-case "$1" in
-
- start)
-    echo -n "Enabling IP masquerading: "
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/10flush mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/10flush
--- e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/10flush	2005-11-20 21:28:08.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/10flush	1969-12-31 17:00:00.000000000 -0700
@@ -1,3 +0,0 @@
-    /sbin/iptables --flush  FORWARD
-    /sbin/iptables --flush  INPUT
-    /sbin/iptables --flush  OUTPUT
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/86startdone mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/86startdone
--- e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/86startdone	2005-11-20 21:28:08.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/86startdone	1969-12-31 17:00:00.000000000 -0700
@@ -1,4 +0,0 @@
-    $0 adjust
-    echo "done"
-    ;;
-
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98restart mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98restart
--- e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98restart	2005-11-20 21:28:08.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98restart	1969-12-31 17:00:00.000000000 -0700
@@ -1,6 +0,0 @@
-restart)
-        $0 stop
-        $0 start
-        ;;
-
-
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98stop mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98stop
--- e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98stop	2005-11-20 21:28:08.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98stop	1969-12-31 17:00:00.000000000 -0700
@@ -1,38 +0,0 @@
-{
-#####STOP FIREWALL####
-}
-stop)
-     echo ""
-     echo -n "Shutting down IP masquerade and firewall rules:"
-     /sbin/iptables -P FORWARD DROP
-     /sbin/iptables -P OUTPUT ACCEPT
-     /sbin/iptables -P INPUT {
-     # Set "safe" default mode.
-     ($SystemMode eq "serveronly") ? "ACCEPT" : "DROP"
-}
-     /sbin/iptables -F INPUT
-     /sbin/iptables -F OUTPUT
-     /sbin/iptables -F FORWARD
-     /sbin/iptables -F 
-{
-    $OUT .= '';
-    # Allow forwarding of local addresses, as we might be a VPN endpoint
-    # in serveronly mode
-    # @locals contains a list of local networks, with the real local
-    # network first
-    my @mylocals = @locals;
-    my $local = shift @mylocals;
-    $OUT .= "    /sbin/iptables --append FORWARD -s $local" .
-			" -d $local -j ACCEPT\n";
-    foreach my $network (@mylocals)
-    {
-	$OUT .= "    /sbin/iptables --append FORWARD -s $network" .
-			" -d $local -j ACCEPT\n";
-	$OUT .= "    /sbin/iptables --append FORWARD -s $local" .
-			" -d $network -j ACCEPT\n";
-    }
-}     /sbin/iptables -X 
-     echo "		Done!"
-     echo "" ;;
-
-
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98trace mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98trace
--- e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98trace	2005-11-20 21:28:08.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98trace	1969-12-31 17:00:00.000000000 -0700
@@ -1,15 +0,0 @@
-trace)
-    trace=$(/sbin/e-smith/config getprop masq Trace)
-    if [ $trace = "enabled" ]; then
-        action="stop"
-        echo "Disabling iptables-trace..."
-        /sbin/e-smith/config setprop masq Trace disabled
-    else
-        action="start"
-        echo "Enabling iptables-trace..."
-        /sbin/e-smith/config setprop masq Trace enabled
-    fi
-
-    /etc/init.d/iptables-trace $action
-    ;;
-
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98usage mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98usage
--- e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98usage	2005-11-20 21:28:08.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates/etc/rc.d/init.d/masq/98usage	1969-12-31 17:00:00.000000000 -0700
@@ -1,6 +0,0 @@
- *)
-    echo "Usage: masq \{start|stop|restart|...\}"
-    exit 1
-
-esac
-exit 0
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.17.2/root/etc/e-smith/templates.metadata/etc/rc.d/init.d/masq mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates.metadata/etc/rc.d/init.d/masq
--- e-smith-base-4.17.2/root/etc/e-smith/templates.metadata/etc/rc.d/init.d/masq	2005-11-20 21:28:10.000000000 -0700
+++ mezzanine_patched_e-smith-base-4.17.2/root/etc/e-smith/templates.metadata/etc/rc.d/init.d/masq	1969-12-31 17:00:00.000000000 -0700
@@ -1 +0,0 @@
-PERMS=0755
