diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/configuration/migrate/10SystemPrimaryDomain mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/migrate/10SystemPrimaryDomain
--- e-smith-base-4.15.8/root/etc/e-smith/db/configuration/migrate/10SystemPrimaryDomain	2005-11-21 15:28:05.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/migrate/10SystemPrimaryDomain	2005-12-14 13:19:29.521554816 +1100
@@ -40,6 +40,7 @@
 			    type => 'domain',
 			    Content => 'Primary',
 			    Description => 'Primary domain',
+			    Nameservers => 'localhost',
 			});
     $domain->merge_props(SystemPrimaryDomain => 'yes', Removable => 'no');
 }
