diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/common/type mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/common/type
--- e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/common/type	2005-11-21 15:28:02.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/common/type	1970-01-01 10:00:00.000000000 +1000
@@ -1 +0,0 @@
-url
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/server-common/type mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/server-common/type
--- e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/server-common/type	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/server-common/type	2005-11-21 15:28:02.000000000 +1100
@@ -0,0 +1 @@
+url
diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/server-resources/type mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/server-resources/type
--- e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/server-resources/type	1970-01-01 10:00:00.000000000 +1000
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/accounts/defaults/server-resources/type	2005-12-14 15:14:04.360410108 +1100
@@ -0,0 +1 @@
+url
