diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/sbin/e-smith/add_mirror mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/add_mirror
--- e-smith-base-4.15.8/root/sbin/e-smith/add_mirror	2005-11-20 23:28:10.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/sbin/e-smith/add_mirror	2005-12-16 14:38:36.000000000 -0500
@@ -68,7 +68,7 @@
 then
     part_check='No partitions found'
 
-    NEW_parts=$(sfdisk -l /dev/$NEW 2>&1 | grep "$part_check")
+    NEW_parts=$(LC_ALL=C sfdisk -l /dev/$NEW 2>&1 | grep "$part_check")
 
     [ "$NEW_parts" = "$part_check" ] || \
         usage "$NEW already has partitions configured"
