diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/db/configuration/migrate/10SetAccessDefaults mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/migrate/10SetAccessDefaults
--- e-smith-base-4.15.8/root/etc/e-smith/db/configuration/migrate/10SetAccessDefaults	2005-11-20 23:28:05.000000000 -0500
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/db/configuration/migrate/10SetAccessDefaults	2006-01-29 21:29:06.000000000 -0500
@@ -22,15 +22,16 @@
 
     my $current_mode = $conf->get_value('SystemMode') or return;
 
-    my $sysconfig = $conf->get("sysconfig") ||
-                $conf->new_record("sysconfig",
-		    {
-			type => "configuration",
-			PreviousSystemMode => $current_mode,
-		    });
+    my $sysconfig = $conf->get("sysconfig") or return;
 
     my $previous_mode = $sysconfig->prop('PreviousSystemMode')
-			    || $current_mode;
+			    || "unknown";
+
+    if ($previous_mode eq "unknown")
+    {
+	$sysconfig->set_prop('PreviousSystemMode', $current_mode);
+	return;
+    }
 
     return if ( $previous_mode eq $current_mode) ;
 
