diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm mezzanine_patched_e-smith-base-4.15.8/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm
--- e-smith-base-4.15.8/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm	2006-01-13 17:02:40.528796550 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/usr/lib/perl5/site_perl/esmith/FormMagick/Panel/useraccounts.pm	2006-01-13 17:02:03.824182740 +1100
@@ -900,7 +900,7 @@
     my ($self) = @_;
     my $acctName = $self->{cgi}->param('acctName');
 
-    unless (($acctName) = ($acctName =~ /^(\w[\-\w]+)$/)) {
+    unless (($acctName) = ($acctName =~ /^(\w[\-\w_\.]+)$/)) {
         return $self->error('TAINTED_USER');
     }
     $acctName = $1;
