diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.8/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/reboot mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/reboot
--- e-smith-base-4.15.8/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/reboot	2005-11-21 15:28:05.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.8/root/etc/e-smith/locale/en-us/etc/e-smith/web/functions/reboot	2005-12-15 17:59:34.740006531 +1100
@@ -23,7 +23,7 @@
     </entry>
     <entry>
 	<base>DESC_REBOOT</base>
-	<trans>In eight seconds, your server will begin the reboot process.</trans>
+	<trans>Your server has initiated the reboot process.</trans>
     </entry>
     <entry>
 	<base>RECONFIGURE</base>
@@ -36,13 +36,13 @@
     <entry>
 	<base>DESC_RECONFIGURE</base>
 	<trans>
-	In eight seconds, your server will perform a full system
+	Your server has initiated a full system
 	reconfiguration and reboot.
     </trans>
     </entry>
     <entry>
 	<base>DESC_SHUTDOWN</base>
-	<trans>In eight seconds, your server will begin the shutdown process. 
+	<trans>Your server has initiated the shutdown process. 
 	    The process will take several minutes to complete, after which
 	    you can switch off the power.</trans>
     </entry>
