diff -Nur -x '*.orig' -x '*.rej' e-smith-base-4.15.9/root/sbin/e-smith/console-menu-items/reboot.pl mezzanine_patched_e-smith-base-4.15.9/root/sbin/e-smith/console-menu-items/reboot.pl
--- e-smith-base-4.15.9/root/sbin/e-smith/console-menu-items/reboot.pl	2006-02-17 14:41:51.000000000 +1100
+++ mezzanine_patched_e-smith-base-4.15.9/root/sbin/e-smith/console-menu-items/reboot.pl	2006-02-21 17:45:40.399497653 +1100
@@ -4,7 +4,7 @@
 {
     my $class = shift;
     my $self = {
-		    name => "Reboot, Reconfigure or shut down this server",
+		    name => "Reboot, reconfigure or shut down this server",
 		    order => 40,
 		};
     bless $self, $class;
